--- a/udev/lircd_helper.in	2015-02-09 11:52:40.887007189 +0000
+++ b/udev/lircd_helper.in	2015-02-09 11:51:02.345653210 +0000
@@ -28,48 +28,72 @@
 #     as well as receiver, then we still need lircd running.
 ################################################################################
 
+logger -t "lircd_helper" ACTION ${ACTION}
+logger -t "lircd_helper" lircd_driver ${lircd_driver}
+logger -t "lircd_helper" DEVNAME ${DEVNAME}
 case "${ACTION}" in
-    add)
+    add|change)
         if test "x${lircd_driver}" = "x" ; then
             exit 1;
         fi
         if test "x${lircd_conf}" = "x" ; then
             exit 1;
         fi
-        if test ! -e '@localstatedir@/run/lirc' ; then
-            mkdir -p '@localstatedir@/run/lirc'
+        if test ! -e '/var/run/lirc' ; then
+            mkdir -p '/var/run/lirc'
         fi
         devname_instance=`echo ${DEVNAME} | sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
-        if test ! -e "@localstatedir@/run/lirc/lircd-${devname_instance}.pid" ; then
-            daemon='@sbindir@/lircd'
+        if test "x${lircd_driver}" = "xatilibusb" ; then
+            devname_instance="atilibusb"
+        fi
+        logger -t "lircd_helper" add/change devname_instance ${devname_instance}
+        if test -e "/var/run/lirc/lircd-${devname_instance}.pid"; then
+            process=$(ps -p $(cat /var/run/lirc/lircd-${devname_instance}.pid) -o comm=)
+            if test ! "x${process}" = "xlircd" ; then
+                logger -t "lircd_helper" removing stale pid "/var/run/lirc/lircd-${devname_instance}.pid"
+                rm -rf "/var/run/lirc/lircd-${devname_instance}.pid"
+            fi
+        fi
+        if test ! -e "/var/run/lirc/lircd-${devname_instance}.pid" ; then
+            daemon='/usr/sbin/lircd'
             daemon="${daemon} --driver=${lircd_driver}"
             daemon="${daemon} --device=${DEVNAME}"
             if test ! "x${lircd_uinput_disable}" = "xtrue" ; then
                 daemon="${daemon} --uinput"
             fi
-            daemon="${daemon} --output=@localstatedir@/run/lirc/lircd-${devname_instance}"
-            daemon="${daemon} --pidfile=@localstatedir@/run/lirc/lircd-${devname_instance}.pid"
+            daemon="${daemon} --output=/var/run/lirc/lircd-${devname_instance}"
+            daemon="${daemon} --pidfile=/var/run/lirc/lircd-${devname_instance}.pid"
             daemon="${daemon} ${lircd_conf}"
-            ${daemon}
+            echo "${daemon}" > "/var/run/lirc/lircd-${devname_instance}.sh"
+            chmod a+x "/var/run/lirc/lircd-${devname_instance}.sh"
+            systemctl start lircd_helper@${devname_instance}.service
+            logger -t "lircd_helper" add/change DEVLINKS ${DEVLINKS}
             for devlink in ${DEVLINKS} ; do
                 devlink_instance=`echo ${devlink} | /bin/sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
-                rm -f "@localstatedir@/run/lirc/lircd-${devlink_instance}"
-                ln -s "lircd-${devname_instance}" "@localstatedir@/run/lirc/lircd-${devlink_instance}"
+                if test ! "x${devlink_instance}" = "x" ; then
+                    logger -t "lircd_helper" add/change devlink_instance ${devlink_instance}
+                    rm -f "/var/run/lirc/lircd-${devlink_instance}"
+                    ln -s "lircd-${devname_instance}" "/var/run/lirc/lircd-${devlink_instance}"
+                fi
             done
         fi
         ;;
     remove)
         instance=`echo $DEVNAME | sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
-        if test -e "@localstatedir@/run/lirc/lircd-${instance}.pid" ; then
-            pid=`cat @localstatedir@/run/lirc/lircd-${instance}.pid`
+        if test "x${lircd_driver}" = "xatilibusb" ; then
+            instance="atilibusb"
+        fi
+        if test -e "/var/run/lirc/lircd-${instance}.pid" ; then
+            pid=`cat /var/run/lirc/lircd-${instance}.pid`
             if test ! "x${pid}" = "x" ; then
-                kill ${pid}
+                systemctl stop lircd_helper@${instance}.service
             fi
             for devlink in ${DEVLINKS} ; do
                 devlink_instance=`echo ${devlink} | sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
-                rm -f "@localstatedir@/run/lirc/lircd-${devlink_instance}"
+                rm -f "/var/run/lirc/lircd-${devlink_instance}"
             done
         fi
+        rm -f "/var/run/lirc/lircd-${instance}.sh"
         ;;
 esac
 
