From 085d17be18190dbdb25597e91048bc27659e07b4 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 4 May 2019 18:54:36 +0100
Subject: [PATCH] Signal when reload should occur from ProfileManager

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/profiles/ProfileManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/profiles/ProfileManager.cpp b/xbmc/profiles/ProfileManager.cpp
index 3b57dcd..70df27d 100644
--- a/xbmc/profiles/ProfileManager.cpp
+++ b/xbmc/profiles/ProfileManager.cpp
@@ -422,7 +422,7 @@ void CProfileManager::FinalizeLoadProfile()
   int firstWindow = g_SkinInfo->GetFirstWindow();
 
   // the startup window is considered part of the initialization as it most likely switches to the final window
-  bool uiInitializationFinished = firstWindow != WINDOW_STARTUP_ANIM;
+  bool uiInitializationFinished = true;
 
   CServiceBroker::GetGUI()->GetWindowManager().ChangeActiveWindow(firstWindow);
 
-- 
2.7.4

