From fcc5131440700be665406ce9c59f6f3fcb9677c4 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 2 Dec 2015 18:13:38 +0000
Subject: [PATCH] Improved user agent for statistic gathering

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/utils/SystemInfo.cpp | 134 +---------------------------------------------
 1 file changed, 1 insertion(+), 133 deletions(-)

diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index 1af6c06..4149216 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -1087,139 +1087,7 @@ std::string CSysInfo::GetHddSpaceInfo(int& percent, int drive, bool shortText)
 std::string CSysInfo::GetUserAgent()
 {
   static std::string result;
-  if (!result.empty())
-    return result;
-
-  result = GetAppName() + "/" + (std::string)g_infoManager.GetLabel(SYSTEM_BUILD_VERSION_SHORT) + " (";
-#if defined(TARGET_WINDOWS)
-  result += GetKernelName() + " " + GetKernelVersion();
-  BOOL bIsWow = FALSE;
-  if (IsWow64Process(GetCurrentProcess(), &bIsWow) && bIsWow)
-      result.append("; WOW64");
-  else
-  {
-    SYSTEM_INFO si = {};
-    GetSystemInfo(&si);
-    if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64)
-      result.append("; Win64; x64");
-    else if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_IA64)
-      result.append("; Win64; IA64");
-    else if (si.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_ARM)
-      result.append("; ARM");
-  }
-#elif defined(TARGET_DARWIN)
-#if defined(TARGET_DARWIN_IOS)
-  std::string iDevStr(GetModelName()); // device model name with number of model version
-  size_t iDevStrDigit = iDevStr.find_first_of("0123456789");
-  std::string iDev(iDevStr, 0, iDevStrDigit);  // device model name without number 
-  if (iDevStrDigit == 0)
-    iDev = "unknown";
-  result += iDev + "; ";
-  std::string iOSVerison(GetOsVersion());
-  size_t lastDotPos = iOSVerison.rfind('.');
-  if (lastDotPos != std::string::npos && iOSVerison.find('.') != lastDotPos
-      && iOSVerison.find_first_not_of('0', lastDotPos + 1) == std::string::npos)
-    iOSVerison.erase(lastDotPos);
-  StringUtils::Replace(iOSVerison, '.', '_');
-  if (iDev == "iPad" || iDev == "AppleTV")
-    result += "CPU OS ";
-  else
-    result += "CPU iPhone OS ";
-  result += iOSVerison + " like Mac OS X";
-#else
-  result += "Macintosh; ";
-  std::string cpuFam(GetBuildTargetCpuFamily());
-  if (cpuFam == "x86")
-    result += "Intel ";
-  else if (cpuFam == "PowerPC")
-    result += "PPC ";
-  result += "Mac OS X ";
-  std::string OSXVersion(GetOsVersion());
-  StringUtils::Replace(OSXVersion, '.', '_');
-  result += OSXVersion;
-#endif
-#elif defined(TARGET_ANDROID)
-  result += "Linux; Android ";
-  std::string versionStr(GetOsVersion());
-  const size_t verLen = versionStr.length();
-  if (verLen >= 2 && versionStr.compare(verLen - 2, 2, ".0", 2) == 0)
-    versionStr.erase(verLen - 2); // remove last ".0" if any
-  result += versionStr;
-  std::string deviceInfo(GetModelName());
-
-  char buildId[PROP_VALUE_MAX];
-  int propLen = __system_property_get("ro.build.id", buildId);
-  if (propLen > 0 && propLen <= PROP_VALUE_MAX)
-  {
-    if (!deviceInfo.empty())
-      deviceInfo += " ";
-    deviceInfo += "Build/";
-    deviceInfo.append(buildId, propLen);
-  }
-
-  if (!deviceInfo.empty())
-    result += "; " + deviceInfo;
-#elif defined(TARGET_POSIX)
-  result += "X11; ";
-  struct utsname un;
-  if (uname(&un) == 0)
-  {
-    std::string cpuStr(un.machine);
-    if (cpuStr == "x86_64" && GetXbmcBitness() == 32)
-      cpuStr = "i686 (x86_64)";
-    result += un.sysname;
-    result += " ";
-    result += cpuStr;
-  }
-  else
-    result += "Unknown";
-#else
-  result += "Unknown";
-#endif
-  result += ")";
-
-  if (GetAppName() != "Kodi")
-    result += " Kodi_Fork_" + GetAppName() + "/1.0"; // default fork number is '1.0', replace it with actual number if necessary
-
-#ifdef TARGET_LINUX
-  // Add distribution name
-  std::string linuxOSName(GetOsName(true));
-  if (!linuxOSName.empty())
-    result += " " + linuxOSName + "/" + GetOsVersion();
-#endif
-
-#ifdef TARGET_RASPBERRY_PI
-  result += " HW_RaspberryPi/1.0";
-#elif defined (TARGET_DARWIN_IOS)
-  std::string iDevVer;
-  if (iDevStrDigit == std::string::npos)
-    iDevVer = "0.0";
-  else
-    iDevVer.assign(iDevStr, iDevStrDigit, std::string::npos);
-  StringUtils::Replace(iDevVer, ',', '.');
-  result += " HW_" + iDev + "/" + iDevVer;
-#endif
-  // add more device IDs here if needed. 
-  // keep only one device ID in result! Form:
-  // result += " HW_" + "deviceID" + "/" + "1.0"; // '1.0' if device has no version
-
-#if defined(TARGET_ANDROID)
-  // Android has no CPU string by default, so add it as additional parameter
-  struct utsname un1;
-  if (uname(&un1) == 0)
-  {
-    std::string cpuStr(un1.machine);
-    StringUtils::Replace(cpuStr, ' ', '_');
-    result += " Sys_CPU/" + cpuStr;
-  }
-#endif
-
-  result += " App_Bitness/" + StringUtils::Format("%d", GetXbmcBitness());
-
-  std::string fullVer(g_infoManager.GetLabel(SYSTEM_BUILD_VERSION));
-  StringUtils::Replace(fullVer, ' ', '-');
-  result += " Version/" + fullVer;
-
+  result = "OSMC " + GetOsVersion() + " running " + GetAppName() + "/" + (std::string)g_infoManager.GetLabel(SYSTEM_BUILD_VERSION_SHORT) + " (Linux)";
   return result;
 }
 
-- 
2.1.0

