From 2b5cb47a05a11f54a6f1a8bb6d76c3fcf2f58280 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 20 May 2016 17:18:13 +0100
Subject: [PATCH] Add enumerated type OSMCSettingsState

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp |  7 +++++++
 xbmc/Application.h   | 12 ++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index f877516..510be06 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -229,6 +229,7 @@ CApplication::CApplication(void)
 #endif
   m_itemCurrentFile(new CFileItem)
   , m_pInertialScrollingHandler(new CInertialScrollingHandler())
+  , m_eOSMCWalkthroughState(OSMC_WALKTHROUGH_NOTRUNNING)
   , m_WaitingExternalCalls(0)
   , m_playerEvent(true, true)
 {
@@ -2650,6 +2651,12 @@ void CApplication::Stop(int exitCode)
   Sleep(200);
 }
 
+void CApplication::SetOSMCWalkthroughState(OSMCWalkthroughState state)
+{
+ m_eOSMCWalkthroughState = state;
+ CServiceBroker::GetRepositoryUpdater().ScheduleUpdate();
+}
+
 bool CApplication::PlayMedia(CFileItem& item, const std::string &player, int iPlaylist)
 {
   //If item is a plugin, expand out
diff --git a/xbmc/Application.h b/xbmc/Application.h
index b38409a..0edd896 100644
--- a/xbmc/Application.h
+++ b/xbmc/Application.h
@@ -157,6 +157,18 @@ public:
 
   bool IsCurrentThread() const;
   void Stop(int exitCode);
+
+  // osmc signals
+
+  typedef enum
+  {
+       OSMC_WALKTHROUGH_NOTRUNNING = 0,
+       OSMC_WALKTHROUGH_ISRUNNING = 1,
+       OSMC_WALKTHROUGH_ISDONE = 2,
+  } OSMCWalkthroughState;
+  OSMCWalkthroughState m_eOSMCWalkthroughState;
+  void SetOSMCWalkthroughState(OSMCWalkthroughState state);
+
   void UnloadSkin(bool forReload = false);
   bool LoadCustomWindows();
   void ReloadSkin(bool confirm = false);
-- 
2.7.4

