From 7f51a2d873e47be1f03b7d9d12f538ab4d357c85 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 23 Aug 2015 17:49:02 +0100
Subject: [PATCH] Add OSMCWalkthroughState to CApplicationMessenger

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp                  | 17 +++++++++++++++++
 xbmc/messaging/ApplicationMessenger.h |  1 +
 2 files changed, 18 insertions(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 510be06..3c1302c 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2110,6 +2110,23 @@ void CApplication::OnApplicationMessage(ThreadMessage* pMsg)
   }
   break;
 
+ case TMSG_SETOSMCWALKTHROUGHSTATE:
+ {
+     switch (pMsg->param1)
+     {
+         case 0:
+             g_application.SetOSMCWalkthroughState(g_application.OSMC_WALKTHROUGH_NOTRUNNING);
+             break;
+         case 1:
+             g_application.SetOSMCWalkthroughState(g_application.OSMC_WALKTHROUGH_ISRUNNING);
+             break;
+         case 2:
+             g_application.SetOSMCWalkthroughState(g_application.OSMC_WALKTHROUGH_ISDONE);
+             break;
+      }
+      break;
+  }
+
   case TMSG_NETWORKMESSAGE:
     m_ServiceManager->GetNetwork().NetworkMessage((CNetwork::EMESSAGE)pMsg->param1, pMsg->param2);
     break;
diff --git a/xbmc/messaging/ApplicationMessenger.h b/xbmc/messaging/ApplicationMessenger.h
index 09101f0..5b312e4 100644
--- a/xbmc/messaging/ApplicationMessenger.h
+++ b/xbmc/messaging/ApplicationMessenger.h
@@ -80,6 +80,7 @@
 #define TMSG_RENDERER_PREINIT             TMSG_MASK_APPLICATION + 31
 #define TMSG_RENDERER_UNINIT              TMSG_MASK_APPLICATION + 32
 #define TMSG_EVENT                        TMSG_MASK_APPLICATION + 33
+#define TMSG_SETOSMCWALKTHROUGHSTATE      TMSG_MASK_APPLICATION + 34
 
 #define TMSG_GUI_INFOLABEL                TMSG_MASK_GUIINFOMANAGER + 0
 #define TMSG_GUI_INFOBOOL                 TMSG_MASK_GUIINFOMANAGER + 1
-- 
2.7.4

