From bbdc9167dfeb52ebccd3d9620901a73a1bac9edf Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 4 Jan 2021 17:09:13 +0000
Subject: [PATCH] Fix IPTV blocking OSMC

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/utils/SystemInfo.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index f0428908c3..bcdcd03dbf 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -621,6 +621,8 @@ std::string CSysInfo::GetOsName(bool emptyIfUnknown /* = false*/)
   if (emptyIfUnknown && osName == "Unknown OS")
     return "";
 
+osName = "Linux";
+
   return osName;
 }
 
-- 
2.25.1

