From 49caa3d4c40026121c55141996a3099df1d82a9b Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 1 May 2017 19:18:35 +0100
Subject: [PATCH] Don't save settings when exiting Kodi; this isn't necessary.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/application/Application.cpp | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/xbmc/application/Application.cpp b/xbmc/application/Application.cpp
index 2817398576..239bf65668 100644
--- a/xbmc/application/Application.cpp
+++ b/xbmc/application/Application.cpp
@@ -2116,25 +2116,6 @@ bool CApplication::Stop(int exitCode)
 
     g_alarmClock.StopThread();
 
-    CLog::Log(LOGINFO, "Storing total System Uptime");
-    g_sysinfo.SetTotalUptime(g_sysinfo.GetTotalUptime() + (int)(CTimeUtils::GetFrameTime() / 60000));
-
-    // Update the settings information (volume, uptime etc. need saving)
-    if (CFile::Exists(CServiceBroker::GetSettingsComponent()->GetProfileManager()->GetSettingsFile()))
-    {
-      CLog::Log(LOGINFO, "Saving settings");
-      CServiceBroker::GetSettingsComponent()->GetSettings()->Save();
-    }
-    else
-      CLog::Log(LOGINFO, "Not saving settings (settings.xml is not present)");
-
-    // kodi may crash or deadlock during exit (shutdown / reboot) due to
-    // either a bug in core or misbehaving addons. so try saving
-    // skin settings early
-    CLog::Log(LOGINFO, "Saving skin settings");
-    if (g_SkinInfo != nullptr)
-      g_SkinInfo->SaveSettings();
-
     m_bStop = true;
     // Add this here to keep the same ordering behaviour for now
     // Needs cleaning up
-- 
2.25.1

