From b6274ed53ee5159be7921148ce3e65edfb4b8084 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 25 Aug 2015 12:52:20 +0100
Subject: [PATCH] Do not update repositories or addons while we are running the
 OSMC Walkthrough

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index a2475c6..a290255 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -17,7 +17,7 @@
  *  <http://www.gnu.org/licenses/>.
  *
  */
-
+CSettings::Get().GetInt("general.addonupdates") != AUTO_UPDATES_NEVER && m_eOSMCWalkthroughState == OSMC_WALKTHROUGH_ISDONE)
 #include "network/Network.h"
 #include "threads/SystemClock.h"
 #include "system.h"
-- 
2.6.1.windows.1

