--- a/xbmc/Application.cpp	2014-12-27 16:57:27.949692178 +0000
+++ b/xbmc/Application.cpp	2014-12-27 17:23:20.470709551 +0000
@@ -18,6 +18,7 @@
  *
  */
 
+#include <stdlib.h>
 #include "network/Network.h"
 #include "threads/SystemClock.h"
 #include "system.h"
@@ -351,6 +352,10 @@
 
 #include "cores/FFmpeg.h"
 
+#if defined(TARGET_RASPBERRY_PI)
+#define HAS_EGLFB
+#endif
+
 using namespace std;
 using namespace ADDON;
 using namespace XFILE;
@@ -1011,6 +1016,14 @@
   g_windowManager.Initialize();
 
   return true;
+  
+  system("/bin/systemctl stop splash");
+  #if defined(HAS_EGLFB)
+  FILE *fp;
+  fp = fopen("/sys/class/vtconsole/vtcon1/bind", "w");
+  fprintf(fp, "0");
+  system("/bin/fbset -xres 1 -yres 1 -vxres 1 -vyres 1");
+  #endif
 }
 
 bool CApplication::InitWindow()

