diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index c61da1b..a76712d 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -240,8 +240,22 @@ bool CAddonInstaller::Install(const std::string &addonID, bool force, const std:
   return false;
 }
 
+bool CAddonInstaller::IsBadAddon(std::string addonName)
+{
+  if (addonName == "plugin.video.genesis" || addonName == "repository.xunitytalk" || addonName == "script.allinone" || addonName == "plugin.video.xunity" || addonName == "plugin.video.1channel" || addonName == "plugin.video.icefilms" || addonName == "plugin.video.phstreams")
+  {
+      return ! CGUIDialogYesNo::ShowAndGetInput("Potentially unwanted addon: " + addonName, "This addon is known to cause issues with OSMC installations.", "We recommend that you do not install it and cannot provide support for it. Do not ask for support in the forum.", "Would you like to continue with installation?", "Cancel Install", "I accept the risk");
+  }
+  return false;
+}
+
+
 bool CAddonInstaller::DoInstall(const AddonPtr &addon, const std::string &hash, bool update, const std::string &referer, bool background)
 {
+  // check whether the addon is dodgy
+  if (IsBadAddon(addon->ID()))
+       return false;
+
   // check whether we already have the addon installing
   CSingleLock lock(m_critSection);
   if (m_downloadJobs.find(addon->ID()) != m_downloadJobs.end())
diff --git a/xbmc/addons/AddonInstaller.h b/xbmc/addons/AddonInstaller.h
index ab93a2f..9853e39 100644
--- a/xbmc/addons/AddonInstaller.h
+++ b/xbmc/addons/AddonInstaller.h
@@ -82,6 +82,8 @@ public:
    */
   bool CheckDependencies(const ADDON::AddonPtr &addon, CAddonDatabase *database = NULL);
 
+  bool IsBadAddon(std::string addonName);
+
   /*! \brief Update all repositories (if needed)
    Runs through all available repositories and queues an update of them if they
    need it (according to the set timeouts) or if forced.  Optionally busy wait
