From 29c61956713c942fd16f29e4a87deeef122ca24d Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Sun, 10 Apr 2016 21:35:12 +0200
Subject: [PATCH] - fix epg search

---
 xbmc/pvr/windows/GUIWindowPVRSearch.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/pvr/windows/GUIWindowPVRSearch.cpp b/xbmc/pvr/windows/GUIWindowPVRSearch.cpp
index f0786f7..e025e10 100644
--- a/xbmc/pvr/windows/GUIWindowPVRSearch.cpp
+++ b/xbmc/pvr/windows/GUIWindowPVRSearch.cpp
@@ -153,12 +153,15 @@ void CGUIWindowPVRSearch::OnPrepareFileItems(CFileItemList &items)
 {
   bool bAddSpecialSearchItem = items.IsEmpty();
 
+  items.Clear();
+
   if (m_bSearchConfirmed)
   {
     m_bSearchConfirmed = false;
 
     bAddSpecialSearchItem = true;
 
+    items.Clear();
     CGUIDialogProgress* dlgProgress = (CGUIDialogProgress*)g_windowManager.GetWindow(WINDOW_DIALOG_PROGRESS);
     if (dlgProgress)
     {
-- 
2.7.4

