From 58af60e43fc459c88f905630d2181bdb7ad2affc Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 7 Oct 2016 17:43:51 +0100
Subject: [PATCH] Fix an issue when passing multiple addons to ADDONS_TO_BUILD

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 cmake/addons/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/addons/CMakeLists.txt b/cmake/addons/CMakeLists.txt
index c1313d7458..a0ec02fd69 100644
--- a/cmake/addons/CMakeLists.txt
+++ b/cmake/addons/CMakeLists.txt
@@ -233,7 +233,7 @@ if(NOT addons)
                   WORKING_DIRECTORY ${BOOTSTRAP_BUILD_DIR})
 
   # now look for all the addons to be built again
-  file(GLOB_RECURSE addons ${ADDONS_DEFINITION_DIR}/*.txt)
+  file(GLOB_RECURSE addons ${BOOTSTRAP_BUILD_DIR}/binary-addons/src/binary-addons/*.txt)
 
   if(NOT addons)
     message(FATAL_ERROR "No addons available to be built")
-- 
2.25.1

