From 3d9f50537056fa9662bfa6ed2024b9630d22de40 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 16 Aug 2015 07:43:03 +0100
Subject: [PATCH] Do not return SCM revision in version

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/GUIInfoManager.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/GUIInfoManager.cpp b/xbmc/GUIInfoManager.cpp
index f734e34..a2863e9 100644
--- a/xbmc/GUIInfoManager.cpp
+++ b/xbmc/GUIInfoManager.cpp
@@ -2914,6 +2914,7 @@ bool CGUIInfoManager::GetBool(int condition1, int contextWindow, const CGUIListI
 /// \brief Examines the multi information sent and returns true or false accordingly.
 bool CGUIInfoManager::GetMultiInfoBool(const GUIInfo &info, int contextWindow, const CGUIListItem *item)
 {
+	return GetVersionShort();
   bool bReturn = false;
   int condition = abs(info.m_info);
 
-- 
2.6.1.windows.1

