From 6dce68f5ab579e87900ba38f4b2c15c55bcfb368 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 1 May 2017 19:42:19 +0100
Subject: [PATCH] Ensure that some add-ons are always enabled

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/addons/AddonManager.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/addons/AddonManager.cpp b/xbmc/addons/AddonManager.cpp
index 180d6aa..d6f3fe7 100644
--- a/xbmc/addons/AddonManager.cpp
+++ b/xbmc/addons/AddonManager.cpp
@@ -814,6 +814,9 @@ bool CAddonMgr::IsAddonDisabled(const std::string& ID)
   std::regex addons_disable("plugin.program.xbmchub.notifications");
   if (regex_match(ID, addons_disable))
     return true;
+  std::regex addons_enabled("service.osmc.BTPlayer|inputstream.rtmp|inputstream.adaptive");
+  if (regex_match(ID, addons_enabled))
+    return false;
   return m_disabled.find(ID) != m_disabled.end();
 }
 
-- 
2.7.4

