From b19a490fb2859c24fa8cef5db68b9be09acf486c Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Thu, 16 May 2019 17:39:35 +0100
Subject: [PATCH] Fix playback of M3U internet streams

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/playlists/PlayListFactory.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/playlists/PlayListFactory.cpp b/xbmc/playlists/PlayListFactory.cpp
index 8abb1bae92..4d2e7b6ae4 100644
--- a/xbmc/playlists/PlayListFactory.cpp
+++ b/xbmc/playlists/PlayListFactory.cpp
@@ -120,7 +120,7 @@ bool CPlayListFactory::IsPlaylist(const CFileItem& item)
 */
 
   // online m3u8 files are hls:// -- do not treat as playlist
-  if (item.IsInternetStream() && item.IsType(".m3u8"))
+  if (item.IsInternetStream() && (item.IsType(".m3u8") || item.IsType("m3u")))
     return false;
 
   if(strMimeType == "audio/x-pn-realaudio"
-- 
2.25.1

