From 212c158edd7455277af1a54d001bcaee2e15abe9 Mon Sep 17 00:00:00 2001
From: Jeffrey Larson <jeffothy@gmail.com>
Date: Mon, 8 Aug 2016 18:15:30 +0000
Subject: [PATCH] Allow volume actions without waking up screensaver

---
 xbmc/input/InputManager.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/xbmc/input/InputManager.cpp b/xbmc/input/InputManager.cpp
index 097245822e..2d0e3e5732 100644
--- a/xbmc/input/InputManager.cpp
+++ b/xbmc/input/InputManager.cpp
@@ -545,7 +545,7 @@ bool CInputManager::HandleKey(const CKey& key)
   g_application.ResetScreenSaver();
 
   // allow some keys to be processed while the screensaver is active
-  if (g_application.WakeUpScreenSaverAndDPMS(processKey) && !processKey)
+  if (!processKey && g_application.WakeUpScreenSaverAndDPMS(processKey))
   {
     CLog::LogF(LOGDEBUG, "%s pressed, screen saver/dpms woken up",
                m_Keyboard.GetKeyName((int)key.GetButtonCode()).c_str());
@@ -682,7 +682,10 @@ bool CInputManager::AlwaysProcess(const CAction& action)
     if (builtInFunction == "powerdown" || builtInFunction == "reboot" ||
         builtInFunction == "restart" || builtInFunction == "restartapp" ||
         builtInFunction == "suspend" || builtInFunction == "hibernate" ||
-        builtInFunction == "quit" || builtInFunction == "shutdown")
+        builtInFunction == "quit" || builtInFunction == "shutdown" 
+        || builtInFunction == "volumeup"
+        || builtInFunction == "volumedown"
+        || builtInFunction == "mute")
     {
       return true;
     }
-- 
2.25.1

