From 6a7582dd26c80689930df02ae86d9100cb75263c Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 2 Jun 2015 18:04:22 +0100
Subject: [PATCH] Make the OSMC skin the default

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/system.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/system.h b/xbmc/system.h
index 5225518..8e8b2f9 100644
--- a/xbmc/system.h
+++ b/xbmc/system.h
@@ -268,5 +268,5 @@
 #if defined(HAS_TOUCH_SKIN) && defined(TARGET_DARWIN_IOS)
 #define DEFAULT_SKIN          "skin.re-touched"
 #else
-#define DEFAULT_SKIN          "skin.confluence"
+#define DEFAULT_SKIN          "skin.osmc"
 #endif
-- 
2.1.0

