From 8fa3b70e063f2b421a27095e950fef99a21cb204 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 8 Oct 2017 18:38:32 +0100
Subject: [PATCH] Ensure that some add-ons are always enabled at Kodi startup

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/addons/AddonDatabase.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/xbmc/addons/AddonDatabase.cpp b/xbmc/addons/AddonDatabase.cpp
index 6de470a9a0..6a6562ea04 100644
--- a/xbmc/addons/AddonDatabase.cpp
+++ b/xbmc/addons/AddonDatabase.cpp
@@ -447,6 +447,15 @@ void CAddonDatabase::SyncInstalled(const std::set<std::string>& ids,
                              id.c_str()));
     }
 
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID LIKE 'script.module.osmcsetting.%%'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='script.module.osmccommon'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='script.module.xmltodict'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='service.osmc.settings'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='skin.osmc'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID == 'inputstream.adaptive'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID == 'inputstream.rtmp'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID LIKE 'resource.language.%%'"));
+
     CommitTransaction();
   }
   catch (...)
-- 
2.25.1

