From 31399599e859166f082b4d1476be7fc230b0ef25 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 23 Aug 2015 22:24:39 +0100
Subject: [PATCH] Allow OSMCWalkthroughState to be set via xbmc Python
 interface

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/interfaces/legacy/ModuleXbmc.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/xbmc/interfaces/legacy/ModuleXbmc.cpp b/xbmc/interfaces/legacy/ModuleXbmc.cpp
index e5b8b40..3b69bc1 100644
--- a/xbmc/interfaces/legacy/ModuleXbmc.cpp
+++ b/xbmc/interfaces/legacy/ModuleXbmc.cpp
@@ -85,6 +85,12 @@ namespace XBMCAddon
       CApplicationMessenger::GetInstance().PostMsg(TMSG_SHUTDOWN);
     }
 
+    void setosmcwalkthroughstatus(int state)
+    {
+      ThreadMessage tMsg = {TMSG_SETOSMCWALKTHROUGHSTATE};
+      tMsg.param1 = state;
+      CApplicationMessenger::Get().SendMessage(tMsg);
+    }
     void restart()
     {
       XBMC_TRACE;
@@ -92,6 +98,7 @@ namespace XBMCAddon
     }
 
     void executescript(const char* script)
+
     {
       XBMC_TRACE;
       if (! script)
-- 
2.1.4

