From 987519e401d50e57db435569b34b562536530ad6 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 23 Aug 2015 15:53:43 +0100
Subject: [PATCH] Add enumerated type OSMCSettingsState

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index d9d15da..315385c 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -268,6 +268,7 @@ CApplication::CApplication(void)
   , m_progressTrackingItem(new CFileItem)
   , m_musicInfoScanner(new CMusicInfoScanner)
   , m_fallbackLanguageLoaded(false)
+  , m_eOSMCWalkthroughState(OSMC_WALKTHROUGH_NOTRUNNING)
 {
   m_network = NULL;
   TiXmlBase::SetCondenseWhiteSpace(false);
-- 
2.1.4

