diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index fc3464d..eb22a90 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -1841,6 +1841,13 @@ bool CApplication::OnSettingsSaving() const
 
 void CApplication::ReloadSkin(bool confirm/*=false*/)
 {
+    FILE *fp;
+    fp = fopen("/tmp/NO_UPDATE", "r");
+    if (fp)
+    {
+		fclose(fp);
+		return;
+    }
   std::string oldSkin = g_SkinInfo ? g_SkinInfo->ID() : "";
 
   CGUIMessage msg(GUI_MSG_LOAD_SKIN, -1, g_windowManager.GetActiveWindow());
