From 120e4ccf7b1f0771f6b8f75d46f961825f5928af Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 5 Feb 2017 01:33:09 +0000
Subject: [PATCH] Disable ADSP

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/ServiceManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/ServiceManager.cpp b/xbmc/ServiceManager.cpp
index 2bbdc49..0d65b5e 100644
--- a/xbmc/ServiceManager.cpp
+++ b/xbmc/ServiceManager.cpp
@@ -80,7 +80,7 @@ bool CServiceManager::Init2()
 
 bool CServiceManager::Init3()
 {
-  m_ADSPManager->Init();
+ // m_ADSPManager->Init();
   m_PVRManager->Init();
   m_contextMenuManager->Init();
   m_gameServices->Init();
-- 
2.7.4

