From 0ef28bb6a6d85cb602760dd43f5120ebe603f049 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Mon, 23 Jan 2017 15:42:22 +0000
Subject: [PATCH] Fix missing enties in patch

---
 xbmc/addons/Addon.cpp | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/xbmc/addons/Addon.cpp b/xbmc/addons/Addon.cpp
index 3b94e5a..0cd38dc 100644
--- a/xbmc/addons/Addon.cpp
+++ b/xbmc/addons/Addon.cpp
@@ -30,6 +30,7 @@
 #include "AddonManager.h"
 #include "addons/settings/AddonSettings.h"
 #include "filesystem/Directory.h"
+#include "filesystem/SpecialProtocol.h"
 #include "filesystem/File.h"
 #include "RepositoryUpdater.h"
 #include "settings/Settings.h"
@@ -332,7 +333,19 @@ std::string CAddon::LibPath() const
 {
   if (m_props.libname.empty())
     return "";
-  return URIUtils::AddFileToFolder(m_addonInfo.Path(), m_addonInfo.LibName()); 
+
+  std::string strLibPath = URIUtils::AddFileToFolder(m_addonInfo.Path(), m_addonInfo.LibName());
+
+  // Check if add-on library has been installed to the binaries path instead
+  std::string strSharePath = CSpecialProtocol::TranslatePath("special://xbmc/");
+  const bool bIsInSharePath = StringUtils::StartsWith(strLibPath, strSharePath);
+  if (bIsInSharePath && !CFile::Exists(strLibPath))
+  {
+    std::string strBinPath = CSpecialProtocol::TranslatePath("special://xbmcbin/");
+    strLibPath.replace(0, strSharePath.length(), strBinPath);
+  }
+
+  return strLibPath;
 }
 
 AddonVersion CAddon::GetDependencyVersion(const std::string &dependencyID) const
-- 
2.1.4

