From cdd947d407112f547639c53a78f3f2c7ad119108 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 25 Aug 2015 12:41:13 +0100
Subject: [PATCH] Allow skin to be changed via xbmc() legacy Python interface

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/interfaces/legacy/ModuleXbmc.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/xbmc/interfaces/legacy/ModuleXbmc.cpp b/xbmc/interfaces/legacy/ModuleXbmc.cpp
index 3b69bc1..f267f0d 100644
--- a/xbmc/interfaces/legacy/ModuleXbmc.cpp
+++ b/xbmc/interfaces/legacy/ModuleXbmc.cpp
@@ -97,6 +97,13 @@ namespace XBMCAddon
       CApplicationMessenger::GetInstance().PostMsg(TMSG_RESTART);
     }
 
+    void setskin(const char* skin)
+    {
+      XBMC_TRACE;
+      if (! skin)
+      return;
+      CSettings::Get().SetString("lookandfeel.skin", skin);
+    }
     void executescript(const char* script)
 
     {
-- 
2.1.4

