From 48785ed6fe15a4d2f7d41d34b773827171349ced Mon Sep 17 00:00:00 2001
From: "Chris \\\"koying\\\" Browet" <cbro@semperpax.com>
Date: Wed, 27 Apr 2016 18:44:34 +0200
Subject: [PATCH] FIX: save skin setting changes immediately

---
 xbmc/settings/SkinSettings.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/settings/SkinSettings.cpp b/xbmc/settings/SkinSettings.cpp
index 71ff616c6c..1f8216c199 100644
--- a/xbmc/settings/SkinSettings.cpp
+++ b/xbmc/settings/SkinSettings.cpp
@@ -50,6 +50,7 @@ const std::string& CSkinSettings::GetString(int setting) const
 void CSkinSettings::SetString(int setting, const std::string &label)
 {
   g_SkinInfo->SetString(setting, label);
+  g_SkinInfo->SaveSettings();
 }
 
 int CSkinSettings::TranslateBool(const std::string &setting)
@@ -70,11 +71,13 @@ int CSkinSettings::GetInt(int setting) const
 void CSkinSettings::SetBool(int setting, bool set)
 {
   g_SkinInfo->SetBool(setting, set);
+  g_SkinInfo->SaveSettings();
 }
 
 void CSkinSettings::Reset(const std::string &setting)
 {
   g_SkinInfo->Reset(setting);
+  g_SkinInfo->SaveSettings();
 }
 
 std::set<ADDON::CSkinSettingPtr> CSkinSettings::GetSettings() const
@@ -96,6 +99,7 @@ std::shared_ptr<const ADDON::CSkinSetting> CSkinSettings::GetSetting(
 void CSkinSettings::Reset()
 {
   g_SkinInfo->Reset();
+  g_SkinInfo->SaveSettings();
 
   CGUIInfoManager& infoMgr = CServiceBroker::GetGUI()->GetInfoManager();
   infoMgr.ResetCache();
-- 
2.25.1

