From 51075c090ac6dd64a70565d5a914b0e25742c8f4 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 20 May 2016 17:18:13 +0100
Subject: [PATCH] Add enumerated type OSMCSettingsState

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp |  7 +++++++
 xbmc/Application.h   | 12 ++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 137bed7..34a0615 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -264,6 +264,7 @@ CApplication::CApplication(void)
   , m_progressTrackingItem(new CFileItem)
   , m_musicInfoScanner(new CMusicInfoScanner)
   , m_fallbackLanguageLoaded(false)
+  , m_eOSMCWalkthroughState(OSMC_WALKTHROUGH_NOTRUNNING)
   , m_WaitingExternalCalls(0)
   , m_ProcessedExternalCalls(0)
 {
@@ -2981,6 +2982,12 @@ void CApplication::Stop(int exitCode)
   Sleep(200);
 }
 
+void CApplication::SetOSMCWalkthroughState(OSMCWalkthroughState state)
+{
+ m_eOSMCWalkthroughState = state;
+ CRepositoryUpdater::GetInstance().ScheduleUpdate();
+}
+
 bool CApplication::PlayMedia(const CFileItem& item, const std::string &player, int iPlaylist)
 {
   //If item is a plugin, expand out now and run ourselves again
diff --git a/xbmc/Application.h b/xbmc/Application.h
index 22a6026..a59f4c5 100644
--- a/xbmc/Application.h
+++ b/xbmc/Application.h
@@ -159,6 +159,18 @@ public:
   void StopPVRManager();
   bool IsCurrentThread() const;
   void Stop(int exitCode);
+
+  // osmc signals
+
+  typedef enum
+  {
+       OSMC_WALKTHROUGH_NOTRUNNING = 0,
+       OSMC_WALKTHROUGH_ISRUNNING = 1,
+       OSMC_WALKTHROUGH_ISDONE = 2,
+  } OSMCWalkthroughState;
+  OSMCWalkthroughState m_eOSMCWalkthroughState;
+  void SetOSMCWalkthroughState(OSMCWalkthroughState state);
+
   void RestartApp();
   void UnloadSkin(bool forReload = false);
   bool LoadUserWindows();
-- 
2.7.4

