From e11ef97e1e6dd73d93fffc4f2f401545eae09751 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 2 Jun 2015 18:07:36 +0100
Subject: [PATCH] Byte-compile addons for additional performance

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/addons/AddonInstaller.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index 0a99d59..0431294 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -94,6 +94,7 @@ void CAddonInstaller::OnJobComplete(unsigned int jobID, bool success, CJob* job)
 
   CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_UPDATE);
   g_windowManager.SendThreadMessage(msg);
+  system("/usr/bin/addon-compiler");
 }
 
 void CAddonInstaller::OnJobProgress(unsigned int jobID, unsigned int progress, unsigned int total, const CJob *job)
-- 
2.1.0

