From a6690503c3de6fc84def5e303b728f90643c7802 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 20 May 2016 16:45:44 +0100
Subject: [PATCH] Do not allow repository updates while we are in the My OSMC
 Walkthrough

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/addons/RepositoryUpdater.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/xbmc/addons/RepositoryUpdater.cpp b/xbmc/addons/RepositoryUpdater.cpp
index c0ed40b..cf14b1c 100644
--- a/xbmc/addons/RepositoryUpdater.cpp
+++ b/xbmc/addons/RepositoryUpdater.cpp
@@ -208,6 +208,11 @@ void CRepositoryUpdater::ScheduleUpdate()
   if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_ADDONS_AUTOUPDATES) == AUTO_UPDATES_NEVER)
     return;
 
+  if (g_application.m_eOSMCWalkthroughState != 2) {
+       CLog::Log(LOGDEBUG, "CRepositoryUpdater: refusing to update until My OSMC asserts");
+       return;
+  }
+
   if (!m_addonMgr.HasAddons(ADDON_REPOSITORY))
     return;
 
-- 
2.7.4

