diff --git a/system/settings/settings.xml b/system/settings/settings.xml
index a20f33a1a0..fdcaa0de18 100755
--- a/system/settings/settings.xml
+++ b/system/settings/settings.xml
@@ -2756,6 +2756,18 @@
           </dependencies>
           <control type="toggle" />
         </setting>
+        <setting id="audiooutput.mixlfe" type="integer" label="36457" help="36458">
+          <level>2</level>
+          <default>0</default>
+          <constraints>
+            <minimum>0</minimum>
+            <step>10</step>
+            <maximum>100</maximum>
+          </constraints>
+          <control type="spinner" format="string">
+            <formatlabel>36459</formatlabel>
+          </control>
+        </setting>
         <setting id="audiooutput.processquality" type="integer" label="13505" help="36169">
           <requirement>HAS_AE_QUALITY_LEVELS</requirement>
           <level>2</level>
diff --git a/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResampleFFMPEG.cpp b/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResampleFFMPEG.cpp
index bfef837114..0badf8b065 100644
--- a/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResampleFFMPEG.cpp
+++ b/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResampleFFMPEG.cpp
@@ -9,6 +9,9 @@
 #include "cores/AudioEngine/Utils/AEUtil.h"
 #include "ActiveAEResampleFFMPEG.h"
 #include "utils/log.h"
+#include "settings/SettingsComponent.h"
+#include "settings/Settings.h"
+#include "ServiceBroker.h"
 
 extern "C" {
 #include <libavutil/channel_layout.h>
@@ -63,6 +66,12 @@ bool CActiveAEResampleFFMPEG::Init(SampleConfig dstConfig, SampleConfig srcConfi
     return false;
   }
 
+  double mix_lfe = (float) CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt("audiooutput.mixlfe") / (float) 100;
+  if (mix_lfe)
+  {
+    av_opt_set_double(m_pContext, "lfe_mix_level", mix_lfe, 0);
+  }
+
   if(quality == AE_QUALITY_HIGH)
   {
     av_opt_set_double(m_pContext, "cutoff", 1.0, 0);
