From 16e8e4c1f8c02ffcfaeafbbc2d80aaade56bb88f Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 29 Oct 2016 02:44:36 +0100
Subject: [PATCH] Backport disable unwanted notifications from 3rd parties to
 Jarvis

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/addons/AddonManager.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/addons/AddonManager.cpp b/xbmc/addons/AddonManager.cpp
index ec60036..accbaff 100644
--- a/xbmc/addons/AddonManager.cpp
+++ b/xbmc/addons/AddonManager.cpp
@@ -52,6 +52,7 @@
 #include "utils/log.h"
 #include "utils/StringUtils.h"
 #include "utils/XBMCTinyXML.h"
+#include <regex>
 
 #ifdef HAS_VISUALISATION
 #include "Visualisation.h"
@@ -685,6 +686,9 @@ bool CAddonMgr::IsBlacklisted(const std::string& id) const
 bool CAddonMgr::DisableAddon(const std::string& id)
 {
   CSingleLock lock(m_critSection);
+  std::regex addons_disable("plugin.program.xbmchub.notifications");
+  if (regex_match(id, addons_disable))
+    return true;
   if (m_disabled.find(id) != m_disabled.end())
     return true; //already disabled
 
-- 
2.7.4

