From 77e860efca65b50606fd2a5b32f4c74286e2d428 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 3 Jun 2015 16:30:41 +0100
Subject: [PATCH] Do not handle mounting in Kodi unless specified in
 advancedsettings

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/settings/AdvancedSettings.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/xbmc/settings/AdvancedSettings.cpp b/xbmc/settings/AdvancedSettings.cpp
index 4fd93d0ccc..67c13ab3f0 100644
--- a/xbmc/settings/AdvancedSettings.cpp
+++ b/xbmc/settings/AdvancedSettings.cpp
@@ -103,8 +103,7 @@ void CAdvancedSettings::Initialize(CSettingsManager& settingsMgr)
   if (params->IsStartFullScreen())
     m_startFullScreen = true;
 
-  if (params->IsStandAlone())
-    m_handleMounting = true;
+  m_handleMounting = false;
 
   settingsMgr.RegisterSettingsHandler(this, true);
   std::set<std::string> settingSet;
-- 
2.25.1

