From db26dd8f619d76cf459b87c2e003e3cd33b96b79 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 8 Jan 2017 20:09:21 +0100
Subject: [PATCH 1/2] Autotools: Fake empty AE override

---
 xbmc/cores/AudioEngine/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/AudioEngine/Makefile.in b/xbmc/cores/AudioEngine/Makefile.in
index 9a4e998..9a3d486 100644
--- a/xbmc/cores/AudioEngine/Makefile.in
+++ b/xbmc/cores/AudioEngine/Makefile.in
@@ -83,6 +83,6 @@ SRCS += Utils/AELimiter.cpp
 SRCS += Encoders/AEEncoderFFmpeg.cpp
 
 LIB   = audioengine.a
-
+$(shell touch AEDefines_override.h)
 include @abs_top_srcdir@/Makefile.include
 -include $(patsubst %.cpp,%.P,$(patsubst %.c,%.P,$(SRCS)))

From 7f0d428dd9b252a69ad11e5a57a9592597e09e94 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 8 Jan 2017 21:39:19 +0100
Subject: [PATCH 2/2] Gitignore: Add AEDefines_override.h for autotools

---
 .gitignore | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index 9c8dc28..c0631de 100644
--- a/.gitignore
+++ b/.gitignore
@@ -564,3 +564,6 @@ exclude_dll.txt
 
 #certificates
 /system/certs/
+
+#Workaround for autotools right before final Krypton
+xbmc/cores/AudioEngine/AEDefines_override.h
