From aae3caa2d1b17b3431434a3208a0edb4d0176419 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 2 Jun 2015 22:23:24 +0100
Subject: [PATCH] Do not decode VC1 or MPEG2 via CrystalHD

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
index 4142c9c..89f0f87 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp
@@ -284,14 +284,9 @@ CDVDVideoCodec* CDVDFactoryCodec::CreateVideoCodec(CDVDStreamInfo &hint, const C
     {
       switch(hint.codec)
       {
-        case AV_CODEC_ID_VC1:
-        case AV_CODEC_ID_WMV3:
         case AV_CODEC_ID_H264:
-        case AV_CODEC_ID_MPEG2VIDEO:
           if (hint.codec == AV_CODEC_ID_H264 && hint.ptsinvalid)
             break;
-          if (hint.codec == AV_CODEC_ID_MPEG2VIDEO && hint.width <= 720)
-            break;
           if ( (pCodec = OpenCodec(new CDVDVideoCodecCrystalHD(), hint, options)) ) return pCodec;
         break;
         default:
-- 
2.1.0

