--- a/xbmc/Application.cpp	2014-12-27 16:57:27.949692178 +0000
+++ b/xbmc/Application.cpp	2014-12-27 20:03:02.521666634 +0000
@@ -351,6 +351,11 @@
 
 #include "cores/FFmpeg.h"
 
+#if defined(TARGET_RASPBERRY_PI)
+#define HAS_EGLFB
+#include <stdlib.h>
+#endif
+
 using namespace std;
 using namespace ADDON;
 using namespace XFILE;
@@ -1011,6 +1016,13 @@
   g_windowManager.Initialize();
 
   return true;
+
+  #if defined(HAS_EGLFB)
+  FILE *fp;
+  fp = fopen("/sys/class/vtconsole/vtcon1/bind", "w");
+  fprintf(fp, "0");
+  system("/usr/bin/sudo /bin/fbset -xres 1 -yres 1 -vxres 1 -vyres 1");
+  #endif
 }
 
 bool CApplication::InitWindow()
