From 598f3d024e0dfa5e7303de49146e70b7f56f56f3 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 15 Sep 2015 05:39:29 +0100
Subject: [PATCH 2/2] Enable libdcadec library

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 tools/depends/target/ffmpeg/autobuild.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/depends/target/ffmpeg/autobuild.sh b/tools/depends/target/ffmpeg/autobuild.sh
index 7804609..ed64e78 100755
--- a/tools/depends/target/ffmpeg/autobuild.sh
+++ b/tools/depends/target/ffmpeg/autobuild.sh
@@ -146,6 +146,7 @@ CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" LDFLAGS="$LDFLAGS" \
 	--disable-libxcb-xfixes \
 	--disable-x11grab \
 	--disable-xlib \
+	--enable-libdcadec \
 	--enable-bzlib \
 	--enable-gnutls \
 	--enable-muxer=spdif \
-- 
2.1.0

