From 16dbdea8e90c9f8af80c8fffcd66fb9913f46258 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 2 Jun 2015 22:01:21 +0100
Subject: [PATCH] Tidy up Raspberry Pi framebuffer after Kodi EGL windowing
 configuration

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp b/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
index c21f23e..e2c453a 100644
--- a/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
@@ -32,6 +32,7 @@
 #include "guilib/StereoscopicsManager.h"
 #include "rendering/RenderSystem.h"
 #include <cassert>
+#include <stdlib.h>
 
 #ifndef __VIDEOCORE4__
 #define __VIDEOCORE4__
@@ -108,6 +109,20 @@ void CEGLNativeTypeRaspberryPI::Initialize()
 
   m_DllBcmHost = new DllBcmHost;
   m_DllBcmHost->Load();
+  CLog::Log(LOGDEBUG, "Unbind vtcon0 to save resources");
+  FILE *fp;
+  fp = fopen("/sys/class/vtconsole/vtcon1/bind", "w");
+  if (fp)
+  {
+      fprintf(fp, "0");
+      fclose(fp);
+      CLog::Log(LOGDEBUG, "Reducing framebuffer resolution with fbset to save resources");
+      system("/bin/fbset -xres 1 -yres 1 -vxres 1 -vyres 1");
+  }
+ else
+ {
+      CLog::Log(LOGDEBUG, "Could not open vtcon1 for writing. You probably ran this independent of the watchdog!");
+ }
 #endif
 }
 
-- 
2.1.0

