From 98ab3e5dbe747fa8ebceb2d8db03db730774ba6e Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 16 Jan 2016 17:14:16 +0000
Subject: [PATCH] Ensure thumb is disabled for hevc optimisations

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 tools/depends/target/ffmpeg/autobuild.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/depends/target/ffmpeg/autobuild.sh b/tools/depends/target/ffmpeg/autobuild.sh
index 4681f47..364ad74 100755
--- a/tools/depends/target/ffmpeg/autobuild.sh
+++ b/tools/depends/target/ffmpeg/autobuild.sh
@@ -176,6 +176,7 @@ CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" LDFLAGS="$LDFLAGS" \
 	--disable-mipsdsp \
 	--disable-mipsdspr2 \
 	--extra-cflags="-DRPI=1" \
+	--disable-thumb \
         ${FLAGS}
 
 make -j ${BUILDTHREADS} 
-- 
2.7.4

