From 22364aa82d72f1b2caa4adc9366fc3395ed49845 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 16 Jan 2016 17:14:16 +0000
Subject: [PATCH] Ensure thumb is disabled for hevc optimisations

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 tools/depends/target/ffmpeg/autobuild.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/depends/target/ffmpeg/autobuild.sh b/tools/depends/target/ffmpeg/autobuild.sh
index 33a39f1..777e17c 100755
--- a/tools/depends/target/ffmpeg/autobuild.sh
+++ b/tools/depends/target/ffmpeg/autobuild.sh
@@ -177,6 +177,7 @@ CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" LDFLAGS="$LDFLAGS" \
 	--disable-mipsdspr1 \
 	--disable-mipsdspr2 \
 	--enable-libdcadec \
+	--disable-thumb \
         ${FLAGS}
 
 make -j ${BUILDTHREADS} 
-- 
2.1.0

