From dde4009e730cee0adc4f870fd2dd4269df78c827 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 27 Dec 2019 23:08:48 +0000
Subject: [PATCH] Fix VC-1 playback

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 .../cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp | 34 ++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index 10e73cc..7cc58a5 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1640,6 +1640,27 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
   am_private->gcodec.ratio64     = am_private->video_ratio64;
   am_private->gcodec.param       = NULL;
 
+  std::list<EINTERLACEMETHOD> methods;
+
+  if (am_private->video_format == VFORMAT_VC1)
+  {
+    /* add deinterlace options */
+    methods.push_back(EINTERLACEMETHOD::VS_INTERLACEMETHOD_NONE);
+    methods.push_back(EINTERLACEMETHOD::VS_INTERLACEMETHOD_DEINTERLACE);
+    methods.push_back(EINTERLACEMETHOD::VS_INTERLACEMETHOD_AUTO);
+    m_processInfo.SetDeinterlacingMethodDefault(EINTERLACEMETHOD::VS_INTERLACEMETHOD_DEINTERLACE);
+    /* engage deinterlacer for progressive streams */
+    SysfsUtils::SetInt("/sys/module/di/parameters/di_debug_flag", 0x10000);
+    SysfsUtils::SetInt("/sys/module/di/parameters/bypass_prog", 0);
+  }
+  else
+  {
+    methods.clear();
+    SysfsUtils::SetInt("/sys/module/di/parameters/di_debug_flag", 0);
+    SysfsUtils::SetInt("/sys/module/di/parameters/bypass_prog", 1);
+  }
+  m_processInfo.UpdateDeinterlacingMethods(methods);
+
   switch(am_private->video_format)
   {
     default:
@@ -1899,6 +1920,19 @@ bool CAMLCodec::AddData(uint8_t *pData, size_t iSize, double dts, double pts)
   if (!m_opened || !pData)
     return false;
 
+  /* stop VC-1 decoder sending two fields */
+  if (am_private->video_format == VFORMAT_VC1)
+  {
+    if (m_processInfo.GetVideoSettings().m_InterlaceMethod == VS_INTERLACEMETHOD_NONE)
+    {
+      SysfsUtils::SetInt("/sys/module/amvdec_vc1/parameters/force_prog", 1);
+    }
+    else
+    {
+      SysfsUtils::SetInt("/sys/module/amvdec_vc1/parameters/force_prog", 0);
+    }
+  }
+
   m_frameSizes.push_back(iSize);
   m_frameSizeSum += iSize;
 
-- 
2.7.4

