From ded9e574e9a405c1383ff47a252b0af57294184f Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 31 Jan 2018 16:12:58 +0000
Subject: [PATCH] SMBFile: read in chunks of 32768K. This fixes SMBv2 issues

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/filesystem/SMBFile.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/xbmc/filesystem/SMBFile.cpp b/xbmc/filesystem/SMBFile.cpp
index 3cfab4c..d0157d4 100644
--- a/xbmc/filesystem/SMBFile.cpp
+++ b/xbmc/filesystem/SMBFile.cpp
@@ -502,8 +502,7 @@ int CSMBFile::Truncate(int64_t size)
 
 ssize_t CSMBFile::Read(void *lpBuf, size_t uiBufSize)
 {
-  if (uiBufSize > SSIZE_MAX)
-    uiBufSize = SSIZE_MAX;
+  uiBufSize = 32768;
 
   if (m_fd == -1)
     return -1;
-- 
2.7.4

