--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2017-04-21 10:59:59.209075420 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2017-04-21 11:06:02.641075420 +0200
@@ -636,7 +643,7 @@
           break;
       }
 
-      if (framerate > 0.0 && (int)m_framerate != (int)framerate)
+      if (framerate > 0.0 && (int)(m_framerate * 100.0) != (int)(framerate * 100.0))
       {
         m_framerate = framerate;
         m_video_rate = (int)(0.5 + (96000.0 / framerate));
