diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
index 501eae5..c4c7b1d 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
@@ -2059,7 +2059,11 @@ void CAMLCodec::Process()
           if (abs_error > 0.150)
           {
             // big error so try to reset pts_pcrscr
-            SetVideoPtsSeconds(app_pts);
+            // do not reset if the error is very big,
+            // might just be dvdplayer wonking out on a discontinuity
+            // and we just need to let it settle down.
+            if (abs_error < 10.0)
+              SetVideoPtsSeconds(app_pts);
           }
           else
           {
