diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
index 04002fd..a288660 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
@@ -1006,8 +1006,10 @@ void CLinuxRendererGLES::ReleaseBuffer(int idx)
   }
 #endif
 #ifdef HAS_IMXVPU
-  if (m_renderMethod & RENDER_IMXMAP)
+  if (m_renderMethod & RENDER_IMXMAP) {
+    g_IMXContext.Blank();
     SAFE_RELEASE(buf.IMXBuffer);
+  }
 #endif
 }
 
