From 1506b155b11c92ec86e539fd6362ff057ce0b58b Mon Sep 17 00:00:00 2001
From: "Chris \\\"Koying\\\" Browet" <cbro@semperpax.com>
Date: Fri, 19 Dec 2014 12:30:04 +0100
Subject: [PATCH 9/9] FIX: [renderer] hide meaningless skips from users

---
 xbmc/cores/VideoRenderers/RenderManager.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoRenderers/RenderManager.cpp b/xbmc/cores/VideoRenderers/RenderManager.cpp
index 7a99ac4..e966c12 100644
--- a/xbmc/cores/VideoRenderers/RenderManager.cpp
+++ b/xbmc/cores/VideoRenderers/RenderManager.cpp
@@ -1196,7 +1196,8 @@ void CXBMCRenderManager::PrepareNextRender()
     while(m_queued.front() != idx)
     {
       requeue(m_discard, m_queued);
-      m_QueueSkip++;
+      if (m_format != RENDER_FMT_BYPASS)  // skips scares users ;)
+        m_QueueSkip++;
     }
 
     m_presentstep   = PRESENT_FLIP;
-- 
2.1.0

