diff --git a/xbmc/powermanagement/PowerManager.cpp b/xbmc/powermanagement/PowerManager.cpp
index 5900f98..45b7c51 100644
--- a/xbmc/powermanagement/PowerManager.cpp
+++ b/xbmc/powermanagement/PowerManager.cpp
@@ -201,11 +201,11 @@ bool CPowerManager::CanPowerdown()
 }
 bool CPowerManager::CanSuspend()
 {
-  return m_instance->CanSuspend();
+  return false;
 }
 bool CPowerManager::CanHibernate()
 {
-  return m_instance->CanHibernate();
+  return false;
 }
 bool CPowerManager::CanReboot()
 {
