From 1c053441edffc6486b384b50ed7340020a33ece0 Mon Sep 17 00:00:00 2001
From: fritsch <peter.fruehberger@gmail.com>
Date: Fri, 10 Apr 2015 20:52:31 +0200
Subject: [PATCH] Only use videoplayer.limitguiupdate on Pi and IMX

---
 xbmc/Application.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 797748d..719af92 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2963,8 +2963,14 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
   {
     m_skipGuiRender = false;
     int fps = 0;
+
+#if defined(TARGET_RASPBERRY_PI) || defined(HAS_IMXVPU)
+    // This code reduces rendering fps of the GUI layer when playing videos in fullscreen mode
+    // it makes only sense on architectures with multiple layers
     if (g_graphicsContext.IsFullScreenVideo() && !m_pPlayer->IsPausedPlayback())
       fps = CSettings::Get().GetInt("videoplayer.limitguiupdate");
+#endif
+
     unsigned int now = XbmcThreads::SystemClockMillis();
     unsigned int frameTime = now - m_lastRenderTime;
     if (fps > 0 && frameTime * fps < 1000)
