diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
index 7236444..501eae5 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
@@ -96,6 +96,11 @@ public:
   virtual int codec_set_cntl_mode(codec_para_t *pcodec, unsigned int mode)=0;
   virtual int codec_set_cntl_avthresh(codec_para_t *pcodec, unsigned int avthresh)=0;
   virtual int codec_set_cntl_syncthresh(codec_para_t *pcodec, unsigned int syncthresh)=0;
+
+  virtual int codec_set_av_threshold(codec_para_t *pcodec, int threshold)=0;
+  virtual int codec_set_video_delay_limited_ms(codec_para_t *pcodec,int delay_ms)=0;
+  virtual int codec_get_video_delay_limited_ms(codec_para_t *pcodec,int *delay_ms)=0;
+  virtual int codec_get_video_cur_delay_ms(codec_para_t *pcodec,int *delay_ms)=0;
 };
 
 class DllLibAmCodec : public DllDynamic, DllLibamCodecInterface
@@ -119,6 +124,11 @@ class DllLibAmCodec : public DllDynamic, DllLibamCodecInterface
   DEFINE_METHOD2(int, codec_set_cntl_avthresh,  (codec_para_t *p1, unsigned int p2))
   DEFINE_METHOD2(int, codec_set_cntl_syncthresh,(codec_para_t *p1, unsigned int p2))
 
+  DEFINE_METHOD2(int, codec_set_av_threshold,   (codec_para_t *p1, int p2))
+  DEFINE_METHOD2(int, codec_set_video_delay_limited_ms, (codec_para_t *p1, int p2))
+  DEFINE_METHOD2(int, codec_get_video_delay_limited_ms, (codec_para_t *p1, int *p2))
+  DEFINE_METHOD2(int, codec_get_video_cur_delay_ms, (codec_para_t *p1, int *p2))
+
   BEGIN_METHOD_RESOLVE()
     RESOLVE_METHOD(codec_init)
     RESOLVE_METHOD(codec_close)
@@ -135,9 +145,21 @@ class DllLibAmCodec : public DllDynamic, DllLibamCodecInterface
     RESOLVE_METHOD(codec_set_cntl_mode)
     RESOLVE_METHOD(codec_set_cntl_avthresh)
     RESOLVE_METHOD(codec_set_cntl_syncthresh)
+
+    RESOLVE_METHOD(codec_set_av_threshold)
   END_METHOD_RESOLVE()
 
 public:
+  bool LoadVideoDelayFunctions()
+  {
+    // added in amcodec/jellybean, manually load them so we do not fault.
+    int rtn;
+    rtn  = m_dll->ResolveExport("codec_set_video_delay_limited_ms", (void**)&m_codec_set_video_delay_limited_ms_ptr, false);
+    rtn &= m_dll->ResolveExport("codec_get_video_delay_limited_ms", (void**)&m_codec_get_video_delay_limited_ms_ptr, false);
+    rtn &= m_dll->ResolveExport("codec_get_video_cur_delay_ms",(void**)&m_codec_get_video_cur_delay_ms_ptr, false);
+    return(rtn == 1);
+  }
+
   void codec_init_para(aml_generic_param *p_in, codec_para_t *p_out)
   {
     memset(p_out, 0x00, sizeof(codec_para_t));
@@ -1392,6 +1414,7 @@ CAMLCodec::CAMLCodec() : CThread("CAMLCodec")
     m_dll->SetFile("libamcodec.so");
     m_dll->Load();
   }
+  m_dll_has_video_delay = m_dll->LoadVideoDelayFunctions();
   am_private->m_dll = m_dll;
   am_private->vcodec.handle             = -1; //init to invalid
   am_private->vcodec.cntl_handle        = -1;
@@ -1616,6 +1639,8 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
   // make sure we are not stuck in pause (amcodec bug)
   m_dll->codec_resume(&am_private->vcodec);
   m_dll->codec_set_cntl_mode(&am_private->vcodec, TRICKMODE_NONE);
+  if (m_dll_has_video_delay)
+    m_dll->codec_set_video_delay_limited_ms(&am_private->vcodec, 1000);
 
   m_dll->codec_set_cntl_avthresh(&am_private->vcodec, AV_SYNC_THRESH);
   m_dll->codec_set_cntl_syncthresh(&am_private->vcodec, 0);
@@ -1707,6 +1732,8 @@ void CAMLCodec::Reset()
   }
   // reset the decoder
   m_dll->codec_reset(&am_private->vcodec);
+  if (m_dll_has_video_delay)
+    m_dll->codec_set_video_delay_limited_ms(&am_private->vcodec, 1000);
   dumpfile_close(am_private);
   dumpfile_open(am_private);
 
@@ -1800,43 +1827,70 @@ int CAMLCodec::Decode(uint8_t *pData, size_t iSize, double dts, double pts)
       // Decoder got stuck; Reset
       Reset();
     }
-
-    // if we seek, then GetTimeSize is wrong as
-    // reports lastpts - cur_pts and hw decoder has
-    // not started outputing new pts values yet.
-    // so we grab the 1st pts sent into driver and
-    // use that to calc GetTimeSize.
-    if (m_1st_pts == 0)
-      m_1st_pts = am_private->am_pkt.lastpts;
+    if (!m_dll_has_video_delay)
+    {
+      // if we seek, then GetTimeSize is wrong as
+      // reports lastpts - cur_pts and hw decoder has
+      // not started outputing new pts values yet.
+      // so we grab the 1st pts sent into driver and
+      // use that to calc GetTimeSize.
+      if (m_1st_pts == 0)
+        m_1st_pts = am_private->am_pkt.lastpts;
+    }
   }
 
-  // if we have still frames, demux size will be small
-  // and we need to pre-buffer more.
-  double target_timesize = 1.0;
-  if (iSize < 20)
-    target_timesize = 2.0;
-
-  // keep hw buffered demux above 1 second
-  if (GetTimeSize() < target_timesize && m_speed == DVD_PLAYSPEED_NORMAL)
-    return VC_BUFFER;
-
-  // wait until we get a new frame or 25ms,
-  if (m_old_pictcnt == m_cur_pictcnt)
-    m_ready_event.WaitMSec(25);
-
   // we must return VC_BUFFER or VC_PICTURE,
   // default to VC_BUFFER.
   int rtn = VC_BUFFER;
-  if (m_old_pictcnt != m_cur_pictcnt)
+  if (m_dll_has_video_delay)
+  {
+    // we must wait here or we can consume
+    // all buffered dvd player video demux packets.
+    // wait long if hw buffers greater than our limit.
+    // wait short if not.
+    int wait_time = 25;
+    if (GetTimeSize() > 1.0)
+      wait_time = 100;
+    m_ready_event.Reset();
+    m_ready_event.WaitMSec(wait_time);
+
+    if (m_old_pictcnt != m_cur_pictcnt)
+    {
+      m_old_pictcnt++;
+      rtn = VC_PICTURE;
+      // we got a new pict, try to keep hw buffered demux around 1 second.
+      if (GetTimeSize() < 1.0)
+        rtn |= VC_BUFFER;
+    }
+  }
+  else
   {
-    m_old_pictcnt++;
-    rtn = VC_PICTURE;
-    // we got a new pict, try and keep hw buffered demux above 2 seconds.
-    // this, combined with the above 1 second check, keeps hw buffered demux between 1 and 2 seconds.
-    // we also check to make sure we keep from filling hw buffer.
-    if (GetTimeSize() < 2.0 && GetDataSize() < m_vbufsize/3)
-      rtn |= VC_BUFFER;
+    // if we have still frames, demux size will be small
+    // and we need to pre-buffer more.
+    double target_timesize = 1.0;
+    if (iSize < 20)
+      target_timesize = 2.0;
+
+    // keep hw buffered demux above 1 second
+    if (GetTimeSize() < target_timesize && m_speed == DVD_PLAYSPEED_NORMAL)
+      return VC_BUFFER;
+
+    // wait until we get a new frame or 25ms,
+    if (m_old_pictcnt == m_cur_pictcnt)
+      m_ready_event.WaitMSec(25);
+
+    if (m_old_pictcnt != m_cur_pictcnt)
+    {
+      m_old_pictcnt++;
+      rtn = VC_PICTURE;
+      // we got a new pict, try and keep hw buffered demux above 2 seconds.
+      // this, combined with the above 1 second check, keeps hw buffered demux between 1 and 2 seconds.
+      // we also check to make sure we keep from filling hw buffer.
+      if (GetTimeSize() < 2.0 && GetDataSize() < m_vbufsize/3)
+        rtn |= VC_BUFFER;
+    }
   }
+
 /*
   CLog::Log(LOGDEBUG, "CAMLCodec::Decode: "
     "rtn(%d), m_cur_pictcnt(%lld), m_cur_pts(%f), lastpts(%f), GetTimeSize(%f), GetDataSize(%d)",
@@ -1866,7 +1920,7 @@ bool CAMLCodec::GetPicture(DVDVideoPicture *pDvdVideoPicture)
   {
     // We are FF/RW; Do not use the Player clock or it just doesn't work
     if (m_cur_pts == 0)
-      pDvdVideoPicture->pts = (double)m_1st_pts / PTS_FREQ * DVD_TIME_BASE;
+      pDvdVideoPicture->pts = (double)am_private->am_pkt.lastpts / PTS_FREQ * DVD_TIME_BASE;
     else
       pDvdVideoPicture->pts = (double)m_cur_pts / PTS_FREQ * DVD_TIME_BASE;
   }
@@ -1922,16 +1976,26 @@ double CAMLCodec::GetTimeSize()
   if (!m_opened)
     return 0;
 
-  // if m_cur_pts is zero, hw decoder was not started yet
-  // so we use the pts of the 1st demux packet that was send
-  // to hw decoder to calc timesize.
-  if (m_cur_pts == 0)
-    m_timesize = (double)(am_private->am_pkt.lastpts - m_1st_pts) / PTS_FREQ;
+  if (m_dll_has_video_delay)
+  {
+    int video_delay_ms;
+    if (m_dll->codec_get_video_cur_delay_ms(&am_private->vcodec, &video_delay_ms) >= 0)
+      m_timesize = (float)video_delay_ms / 1000.0;
+  }
   else
-    m_timesize = (double)(am_private->am_pkt.lastpts - m_cur_pts) / PTS_FREQ;
+  {
+    // if m_cur_pts is zero, hw decoder was not started yet
+    // so we use the pts of the 1st demux packet that was send
+    // to hw decoder to calc timesize.
+    if (m_cur_pts == 0)
+      m_timesize = (double)(am_private->am_pkt.lastpts - m_1st_pts) / PTS_FREQ;
+    else
+      m_timesize = (double)(am_private->am_pkt.lastpts - m_cur_pts) / PTS_FREQ;
+  }
 
   // lie to DVDPlayer, it is hardcoded to a max of 8 seconds,
-  // if you buffer more than 8 seconds, it goes nuts.
+  // if you buffer more than 8 seconds, it goes nuts. We need
+  // real value in CAMLCodec::Decode so return a local value.
   double timesize = m_timesize;
   if (timesize < 0.0)
     timesize = 0.0;
@@ -1962,7 +2026,7 @@ void CAMLCodec::Process()
       }
 
       pts_video = get_pts_video();
-      if (m_cur_pts != pts_video)
+      if (m_cur_pts != pts_video && pts_video != 0)
       {
         //CLog::Log(LOGDEBUG, "CAMLCodec::Process: pts_video(%lld), pts_video/PTS_FREQ(%f), duration(%f)",
         //  pts_video, (double)pts_video/PTS_FREQ, 1.0/((double)(pts_video - m_cur_pts)/PTS_FREQ));
diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.h b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.h
index 3aa025d..dc58527 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.h
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.h
@@ -66,6 +66,7 @@ private:
   static void   RenderUpdateCallBack(const void *ctx, const CRect &SrcRect, const CRect &DestRect);
 
   DllLibAmCodec   *m_dll;
+  bool             m_dll_has_video_delay;
   bool             m_opened;
   am_private_t    *am_private;
   CDVDStreamInfo   m_hints;
