From 408be8f8ab3cad4d96ecc61f8d13b4dd8a1fd52d Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 13 Feb 2016 04:32:53 +0000
Subject: [PATCH] Hacky way to enable passthrough on Vero 2

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp | 35 +++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index f684c53..afbcb22 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -27,6 +27,7 @@
 #include "settings/SettingsComponent.h"
 #if defined(HAS_LIBAMCODEC)
 #include "utils/AMLUtils.h"
+#include "utils/SysfsUtils.h"
 #endif
 
 #ifdef TARGET_POSIX
@@ -506,6 +507,40 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
   if (aml_present())
   {
     aml_set_audio_passthrough(m_passthrough);
+
+    int aml_digital_codec;
+    switch(format.m_dataFormat)
+    {
+      case CAEStreamInfo::STREAM_TYPE_AC3:
+        aml_digital_codec = 2;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_DTSHD_CORE:
+      case CAEStreamInfo::STREAM_TYPE_DTS_1024:
+      case CAEStreamInfo::STREAM_TYPE_DTS_2048:
+      case CAEStreamInfo::STREAM_TYPE_DTS_512:
+        aml_digital_codec = 3;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_EAC3:
+        aml_digital_codec = 4;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_DTSHD:
+        aml_digital_codec = 5;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_TRUEHD:
+        aml_digital_codec = 7;
+        break;
+
+      default:
+        if (inconfig.channels > 2)
+          aml_digital_codec = 6;
+        else
+          aml_digital_codec = 0;
+    }
+    SysfsUtils::SetInt("/sys/class/audiodsp/digital_codec", aml_digital_codec);
   }
 #endif
 
-- 
2.7.4

