From df3ad93253c7e9a850ec466b31f12ca15f3c59fa Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Fri, 10 Apr 2015 13:05:39 +0100
Subject: [PATCH 2/2] Logging of the i.MX window bringup

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/windowing/egl/EGLNativeTypeIMX.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index 8d67bcc..e1a6b27 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -244,14 +244,28 @@ bool CEGLNativeTypeIMX::SetNativeResolution(const RESOLUTION_INFO &res)
 
   std::string mode;
   SysfsUtils::GetString("/sys/class/graphics/fb0/mode", mode);
+  CLog::Log(LOGDEBUG, "%s%s%s%s", "res.strID = ", res.strId.c_str(), " and mode is ", mode.c_str());
   if (res.strId == mode)
+  {
+	CLog::Log(LOGDEBUG, "%s", "Not going to SetNativeResolution, because mode seems to already be matching");
     return false;
+  }
 
   DestroyNativeWindow();
   DestroyNativeDisplay();
 
   SysfsUtils::SetString("/sys/class/graphics/fb0/mode", res.strId + "\n");
-
+  SysfsUtils::GetString("/sys/class/graphics/fb0/mode", mode);
+  if (res.strId == mode)
+	CLog::Log(LOGDEBUG, "%s", "Mode was changed successfully");
+  else
+  {
+	CLog::Log(LOGDEBUG, "%s", "Mode was not changed successfully");
+	CLog::Log(LOGDEBUG, "%s%s", "fb0 mode is ", mode.c_str());
+	CLog::Log(LOGDEBUG, "%s", "Going to chown to osmc user and try again..");
+	system("/usr/bin/sudo /bin/chown osmc:osmc /sys/class/graphics/fb0/mode");
+	SysfsUtils::SetString("/sys/class/graphics/fb0/mode", res.strId + "\n");
+  }
   CreateNativeDisplay();
 
   CLog::Log(LOGDEBUG, "%s: %s",__FUNCTION__, res.strId.c_str());
-- 
2.1.0

