diff --git a/addons/resource.language.en_gb/resources/strings.po b/addons/resource.language.en_gb/resources/strings.po
index a697a61..0cc9e09 100644
--- a/addons/resource.language.en_gb/resources/strings.po
+++ b/addons/resource.language.en_gb/resources/strings.po
@@ -7961,7 +7961,7 @@ msgstr ""
 
 #: xbmc/video/dialogs/GUIDialogVideoSettings.cpp
 msgctxt "#16020"
-msgid "Deinterlace"
+msgid "Yadif"
 msgstr ""
 
 #: xbmc/video/dialogs/GUIDialogVideoSettings.cpp
@@ -8333,7 +8333,12 @@ msgctxt "#16335"
 msgid "IMX - Fast motion (double)"
 msgstr ""
 
-#empty strings from id 16336 to 16399
+#: xbmc/video/dialogs/GUIDialogVideoSettings.cpp
+msgctxt "#16336"
+msgid "Software - FFmpeg"
+msgstr ""
+
+#empty strings from id 16337 to 16399
 
 #: xbmc/video/dialogs/GUIDialogVideoSettings.cpp
 msgctxt "#16400"
diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
index 1c4715a..9e11c71 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
@@ -3009,15 +3009,12 @@ bool CLinuxRendererGLES::Supports(EINTERLACEMETHOD method)
       return false;
   }
 
-#if !defined(TARGET_ANDROID) && (defined(__i386__) || defined(__x86_64__))
-  if(method == VS_INTERLACEMETHOD_DEINTERLACE
-  || method == VS_INTERLACEMETHOD_DEINTERLACE_HALF
-  || method == VS_INTERLACEMETHOD_SW_BLEND)
-#else
-  if(method == VS_INTERLACEMETHOD_SW_BLEND
+  if(method == VS_INTERLACEMETHOD_YADIF
+  || method == VS_INTERLACEMETHOD_YADIF_HALF
   || method == VS_INTERLACEMETHOD_RENDER_BOB
+  || method == VS_INTERLACEMETHOD_SW_FFMPEG
+  || method == VS_INTERLACEMETHOD_SW_BLEND
   || method == VS_INTERLACEMETHOD_RENDER_BOB_INVERTED)
-#endif
     return true;
 
   return false;
@@ -3074,11 +3071,7 @@ EINTERLACEMETHOD CLinuxRendererGLES::AutoInterlaceMethod()
   if(m_renderMethod & RENDER_IMXMAP)
     return VS_INTERLACEMETHOD_IMX_FASTMOTION;
 
-#if !defined(TARGET_ANDROID) && (defined(__i386__) || defined(__x86_64__))
-  return VS_INTERLACEMETHOD_DEINTERLACE_HALF;
-#else
-  return VS_INTERLACEMETHOD_RENDER_BOB;
-#endif
+  return VS_INTERLACEMETHOD_YADIF_HALF;
 }
 
 CRenderInfo CLinuxRendererGLES::GetRenderInfo()
diff --git a/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp b/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
index 379c541..b4de66c 100644
--- a/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
@@ -553,9 +553,9 @@ void CDVDPlayerVideo::Process()
 
       if (mDeintMode != VS_DEINTERLACEMODE_OFF)
       {
-        if (mInt == VS_INTERLACEMETHOD_DEINTERLACE)
+        if (mInt == VS_INTERLACEMETHOD_YADIF)
           mFilters = CDVDVideoCodec::FILTER_DEINTERLACE_ANY;
-        else if(mInt == VS_INTERLACEMETHOD_DEINTERLACE_HALF)
+        else if(mInt == VS_INTERLACEMETHOD_YADIF_HALF)
           mFilters = CDVDVideoCodec::FILTER_DEINTERLACE_ANY | CDVDVideoCodec::FILTER_DEINTERLACE_HALFED;
 
         if (mDeintMode == VS_DEINTERLACEMODE_AUTO && mFilters)
@@ -678,6 +678,13 @@ void CDVDPlayerVideo::Process()
                 sPostProcessType += g_advancedSettings.m_videoPPFFmpegDeint;
                 bPostProcessDeint = true;
               }
+              else if(mInt == VS_INTERLACEMETHOD_SW_FFMPEG)
+              {
+                if (!sPostProcessType.empty())
+                  sPostProcessType += ",";
+                sPostProcessType += "ffmpegdeint";
+                bPostProcessDeint = true;
+              }
             }
 
             if (CMediaSettings::GetInstance().GetCurrentVideoSettings().m_PostProcess)
diff --git a/xbmc/settings/VideoSettings.h b/xbmc/settings/VideoSettings.h
index 0305058..aa30dc6 100644
--- a/xbmc/settings/VideoSettings.h
+++ b/xbmc/settings/VideoSettings.h
@@ -45,7 +45,7 @@ enum EINTERLACEMETHOD
   VS_INTERLACEMETHOD_RENDER_BOB_INVERTED=5,
   VS_INTERLACEMETHOD_RENDER_BOB=6,
 
-  VS_INTERLACEMETHOD_DEINTERLACE=7,
+  VS_INTERLACEMETHOD_YADIF=7,
 
   VS_INTERLACEMETHOD_VDPAU_BOB=8,
   VS_INTERLACEMETHOD_INVERSE_TELECINE=9,
@@ -55,7 +55,7 @@ enum EINTERLACEMETHOD
   VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF=13,
   VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL=14,
   VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL_HALF=15,
-  VS_INTERLACEMETHOD_DEINTERLACE_HALF=16,
+  VS_INTERLACEMETHOD_YADIF_HALF=16,
   VS_INTERLACEMETHOD_DXVA_BOB = 17,
   VS_INTERLACEMETHOD_DXVA_BEST = 18,
   // VS_INTERLACEMETHOD_DXVA_ANY = 19, Legacy
@@ -75,6 +75,8 @@ enum EINTERLACEMETHOD
   VS_INTERLACEMETHOD_IMX_FASTMOTION = 29,
   VS_INTERLACEMETHOD_IMX_FASTMOTION_DOUBLE = 30,
 
+  VS_INTERLACEMETHOD_SW_FFMPEG = 31,
+
   VS_INTERLACEMETHOD_MAX // do not use and keep as last enum value.
 };
 
diff --git a/xbmc/video/dialogs/GUIDialogVideoSettings.cpp b/xbmc/video/dialogs/GUIDialogVideoSettings.cpp
index b248566..f50307c 100644
--- a/xbmc/video/dialogs/GUIDialogVideoSettings.cpp
+++ b/xbmc/video/dialogs/GUIDialogVideoSettings.cpp
@@ -252,9 +252,10 @@ void CGUIDialogVideoSettings::InitializeSettings()
   entries.push_back(std::make_pair(20129, VS_INTERLACEMETHOD_RENDER_WEAVE));
   entries.push_back(std::make_pair(16022, VS_INTERLACEMETHOD_RENDER_BOB_INVERTED));
   entries.push_back(std::make_pair(16021, VS_INTERLACEMETHOD_RENDER_BOB));
-  entries.push_back(std::make_pair(16020, VS_INTERLACEMETHOD_DEINTERLACE));
-  entries.push_back(std::make_pair(16036, VS_INTERLACEMETHOD_DEINTERLACE_HALF));
+  entries.push_back(std::make_pair(16020, VS_INTERLACEMETHOD_YADIF));
+  entries.push_back(std::make_pair(16036, VS_INTERLACEMETHOD_YADIF_HALF));
   entries.push_back(std::make_pair(16324, VS_INTERLACEMETHOD_SW_BLEND));
+  entries.push_back(std::make_pair(16336, VS_INTERLACEMETHOD_SW_FFMPEG));
   entries.push_back(std::make_pair(16314, VS_INTERLACEMETHOD_INVERSE_TELECINE));
   entries.push_back(std::make_pair(16311, VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL));
   entries.push_back(std::make_pair(16310, VS_INTERLACEMETHOD_VDPAU_TEMPORAL));
