diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
index 8f2e516..2e22fb7 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
@@ -211,6 +211,8 @@ void CDVDVideoCodecAmlogic::Dispose(void)
 {
   if (m_Codec)
     m_Codec->CloseDecoder(), delete m_Codec, m_Codec = NULL;
+  m_opened = false;
+
   if (m_videobuffer.iFlags)
     m_videobuffer.iFlags = 0;
   if (m_mpeg2_sequence)
