From 5a9f3552e0a2db0a3bb4d4ab973ceb660fcaa350 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 14 Jun 2015 22:43:59 +0100
Subject: [PATCH] Disable libXCB when building ffmpeg

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 tools/depends/target/ffmpeg/autobuild.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/depends/target/ffmpeg/autobuild.sh b/tools/depends/target/ffmpeg/autobuild.sh
index 196c347..204813e 100755
--- a/tools/depends/target/ffmpeg/autobuild.sh
+++ b/tools/depends/target/ffmpeg/autobuild.sh
@@ -140,6 +140,7 @@ CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" LDFLAGS="$LDFLAGS" \
 	--enable-postproc \
 	--enable-vaapi \
 	--enable-vdpau \
+	--disable-libxcb \
 	--enable-bzlib \
 	--enable-gnutls \
 	--enable-muxer=spdif \
-- 
2.1.0

