From 6dd7fd889c95f24377b8503fd8cc7bcab4bb89bd Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 14 Oct 2015 16:34:30 +0100
Subject: [PATCH] Better handling of case when SAR is not sane.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/windowing/egl/EGLNativeTypeIMX.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index c9bd48a..923480c 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -409,7 +409,14 @@ void CEGLNativeTypeIMX::GetMonitorSAR()
   // if it is 0, EDID info was missing or calculated
   // SAR value wasn't sane
   if (m_sar == 0)
+  {
     CLog::Log(LOGDEBUG, "%s: Screen SAR - not usable info",__FUNCTION__);
+    RESOLUTION_INFO res = CDisplaySettings::Get().GetCurrentResolutionInfo();
+    if (res.iScreenWidth != 0)
+      m_sar = res.iScreenHeight / res.iScreenWidth;
+    else
+      m_sar = .0f;
+  }
 }
 
 inline double roundf_n(double dval, int n)
-- 
2.1.0

