From 59249c3a55f721e1771458a3d0a8f051e7ec6760 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 24 Jan 2017 16:20:03 +0000
Subject: [PATCH] Allow native build of Kodi on ARM64

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 m4/xbmc_arch.m4 | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/m4/xbmc_arch.m4 b/m4/xbmc_arch.m4
index d28f263..4c7d8dd 100644
--- a/m4/xbmc_arch.m4
+++ b/m4/xbmc_arch.m4
@@ -26,6 +26,9 @@ case $build in
   arm*-*-linux-gnu*|arm*-*-linux-uclibc*)
      AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
      ;;
+  aarch64-*-linux-gnu*)
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
+     ;;
   *)
      AC_MSG_ERROR(unsupported native build platform: $build)
 esac
-- 
2.7.4

