From fede93f3463bc8542000ffd231693cab2b6fc77e Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 3 Jun 2015 16:09:50 +0100
Subject: [PATCH] Hide Vero serial number and temperature

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/windows/GUIWindowSystemInfo.cpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/xbmc/windows/GUIWindowSystemInfo.cpp b/xbmc/windows/GUIWindowSystemInfo.cpp
index 3bde0c9181..da81377627 100644
--- a/xbmc/windows/GUIWindowSystemInfo.cpp
+++ b/xbmc/windows/GUIWindowSystemInfo.cpp
@@ -149,11 +149,7 @@ void CGUIWindowSystemInfo::FrameMove()
     SET_CONTROL_LABEL(i++, g_sysinfo.GetCPUBogoMips());
     if (!g_sysinfo.GetCPUSoC().empty())
       SET_CONTROL_LABEL(i++, g_sysinfo.GetCPUSoC());
-    SET_CONTROL_LABEL(i++, g_sysinfo.GetCPUHardware());
-    SET_CONTROL_LABEL(i++, g_sysinfo.GetCPURevision());
-    SET_CONTROL_LABEL(i++, g_sysinfo.GetCPUSerial());
 #endif
-    SetControlLabel(i++, "%s %s", 22011, SYSTEM_CPU_TEMPERATURE);
 #if (!defined(__arm__) && !defined(__aarch64__)) || defined(TARGET_RASPBERRY_PI)
     SetControlLabel(i++, "%s %s", 13284, SYSTEM_CPUFREQUENCY);
 #endif
-- 
2.16.4
