From: afl1
Date: 2017/10/5
Subject: [PATCH] VideoPlayer: fps fix

--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2017-04-29 09:01:26.783510000 +0200
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2017-04-29 09:50:16.543658000 +0200
@@ -3583,6 +3583,10 @@
 
     if(m_pInputStream && m_pInputStream->IsStreamType(DVDSTREAM_TYPE_DVD))
       hint.filename = "dvd";
+    if (hint.fpsrate / hint.fpsscale > 200) {
+      hint.fpsrate = 60000;
+      hint.fpsscale = 1001;
+    }
   }
   else if(STREAM_SOURCE_MASK(source) == STREAM_SOURCE_VIDEOMUX)
   {
   if (!OpenAmlVideo(hints))
--
