From 5e95cf60793494cd7648597d58bfae60a3985490 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 27 Jul 2016 18:17:32 +0100
Subject: [PATCH] Add support for 800p and 1200p modes

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/utils/AMLUtils.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index f8b3381..250b5db 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -616,6 +616,24 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 60;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
+  else if (StringUtils::EqualsNoCase(fromMode, "800p"))
+  {
+    res->iWidth = 800;
+    res->iHeight= 1200;
+    res->iScreenWidth = 1600;
+    res->iScreenHeight= 2560;
+    res->fRefreshRate = 60;
+    res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
+  }
+  else if (StringUtils::EqualsNoCase(fromMode, "1200p"))
+  {
+    res->iWidth = 1200;
+    res->iHeight= 1920;
+    res->iScreenWidth = 2400;
+    res->iScreenHeight= 3840;
+    res->fRefreshRate = 60;
+    res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
+  }
   else
   {
     return false;
-- 
2.7.4

