From 6bb686f4c0d8b50b27340f0b1f25ccbdca42c144 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Thu, 27 Nov 2014 16:31:56 +0000
Subject: [PATCH] Reduce priority of python threads

---
 xbmc/interfaces/generic/LanguageInvokerThread.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/interfaces/generic/LanguageInvokerThread.cpp b/xbmc/interfaces/generic/LanguageInvokerThread.cpp
index c1000e2..92f2f70 100644
--- a/xbmc/interfaces/generic/LanguageInvokerThread.cpp
+++ b/xbmc/interfaces/generic/LanguageInvokerThread.cpp
@@ -55,6 +55,9 @@ bool CLanguageInvokerThread::execute(const std::string &script, const std::vecto
 
   //Todo wait until running
 
+  /* low prio */
+  SetPriority(GetPriority()-1);
+
   return true;
 }
 
-- 
2.7.4

