From 1228e6c6c49d7a225737c097dfe05b2aef10d4b4 Mon Sep 17 00:00:00 2001
From: tanio99 <tanio99@wolke7.net>
Date: Wed, 21 Aug 2019 10:05:26 +0200
Subject: [PATCH] AMLUtils: toggle TMDS during resolution set

---
 xbmc/utils/AMLUtils.cpp | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index 4c36f6d..3471022 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -484,6 +484,13 @@ bool aml_get_native_resolution(RESOLUTION_INFO *res)
 
 static void aml_enable_PHY(bool enable)
 {
+    int phy_delay = 0;
+    char *phy_env = getenv("phydelay");
+    if (phy_env != NULL) {
+        phy_delay = atoi(phy_env);
+    }
+    CLog::Log(LOGDEBUG, "Waiting %d milliseconds for PHY toggling", phy_delay);
+    usleep(phy_delay * 1000);
     SysfsUtils::SetString("/sys/class/amhdmitx/amhdmitx0/phy", enable ? "1" : "0");
 }
 
@@ -491,11 +498,14 @@ bool aml_set_native_resolution(const RESOLUTION_INFO &res, std::string framebuff
 {
   bool result = false;
 
-  aml_handle_display_stereo_mode(RENDER_STEREO_MODE_OFF);
-  result = aml_set_display_resolution(res, framebuffer_name);
+  aml_enable_PHY(false);
 
-  aml_handle_scale(res);
   aml_handle_display_stereo_mode(stereo_mode);
+  result = aml_set_display_resolution(res, framebuffer_name);
+  aml_handle_scale(res);
+
+  aml_enable_PHY(true);
+
 
   return result;
 }
-- 
2.7.4

