From aee622ea1ecd8e45bcd4de1e83aaa45f2af17bd5 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 13 Apr 2019 21:07:37 +0100
Subject: [PATCH] Turn off 3D when switching resolution. Thanks @kszaq

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/utils/AMLUtils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index fe1dae0..24dabc3 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -474,6 +474,7 @@ bool aml_set_native_resolution(const RESOLUTION_INFO &res, std::string framebuff
 {
   bool result = false;
 
+  aml_handle_display_stereo_mode(RENDER_STEREO_MODE_OFF);
   result = aml_set_display_resolution(res, framebuffer_name);
 
   aml_handle_scale(res);
-- 
2.7.4

