From 35d6ddc68e75e2c094eaceccbbe3432d23b674bb Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 6 Nov 2017 18:11:02 +0000
Subject: [PATCH] Fix missing include for stat.h

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 src/helpers/job-drive-detach.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/helpers/job-drive-detach.c b/src/helpers/job-drive-detach.c
index eeafcab..f073362 100644
--- a/src/helpers/job-drive-detach.c
+++ b/src/helpers/job-drive-detach.c
@@ -30,6 +30,7 @@
 #include <libudev.h>
 
 #include <glib.h>
+#include <sys/stat.h>
 
 static void
 usage (void)
-- 
2.7.4

