--- a/xbmc/windows/GUIWindowSystemInfo.cpp	2013-01-28 16:03:17.917628076 +0000
+++ b/xbmc/windows/GUIWindowSystemInfo.cpp	2013-01-28 16:07:11.797668658 +0000
@@ -56,7 +56,7 @@
   case GUI_MSG_WINDOW_INIT:
     {
       CGUIWindow::OnMessage(message);
+      SET_CONTROL_LABEL(52, "OSMC running XBMC " + g_infoManager.GetLabel(SYSTEM_BUILD_VERSION) +
-      SET_CONTROL_LABEL(52, "XBMC " + g_infoManager.GetLabel(SYSTEM_BUILD_VERSION) +
                             " (Compiled: " + g_infoManager.GetLabel(SYSTEM_BUILD_DATE)+")");
       CONTROL_ENABLE_ON_CONDITION(CONTROL_BT_PVR,
                                   PVR::CPVRManager::Get().IsStarted());

