--- a/package/dosfstools/dosfstools.mk	2021-01-14 05:29:35.385847314 +0000
+++ b/package/dosfstools/dosfstools.mk	2021-01-14 05:30:23.878227689 +0000
@@ -6,7 +6,7 @@
 
 DOSFSTOOLS_VERSION = 3.0.26
 DOSFSTOOLS_SOURCE = dosfstools-$(DOSFSTOOLS_VERSION).tar.xz
-DOSFSTOOLS_SITE = http://daniel-baumann.ch/files/software/dosfstools
+DOSFSTOOLS_SITE = https://download.osmc.tv/buildroot/2014.05
 DOSFSTOOLS_LICENSE = GPLv3+
 DOSFSTOOLS_LICENSE_FILES = COPYING
 
