--- a/package/gcc/Config.in.host	2020-11-16 23:03:34.804263415 +0000
+++ b/package/gcc/Config.in.host	2020-11-16 23:03:48.908772847 +0000
@@ -11,7 +11,7 @@
 	default BR2_GCC_VERSION_4_8_ARC if BR2_arc
 	default BR2_GCC_VERSION_4_9_X if BR2_microblaze
 	default BR2_GCC_VERSION_4_5_X if BR2_bfin
-	default BR2_GCC_VERSION_4_7_X
+	default BR2_GCC_VERSION_4_9_X
 	help
 	  Select the version of gcc you wish to use.
 
