--- a/package/pkgconf/pkgconf.mk	2021-10-10 14:10:28.327109209 +0100
+++ b/package/pkgconf/pkgconf.mk	2021-10-10 14:10:50.595358834 +0100
@@ -5,7 +5,7 @@
 ################################################################################
 
 PKGCONF_VERSION = 0.8.9
-PKGCONF_SITE = http://rabbit.dereferenced.org/~nenolod/distfiles/
+PKGCONF_SITE = http://download.osmc.tv/buildroot/2014.05
 PKGCONF_SOURCE = pkgconf-$(PKGCONF_VERSION).tar.bz2
 PKGCONF_LICENSE = pkgconf license
 PKGCONF_LICENSE_FILES = COPYING
