--- b/package/automake/automake.mk	2019-04-22 18:55:25.525608073 +0100
+++ b/package/automake/automake.mk	2019-04-22 18:55:33.545636039 +0100
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-AUTOMAKE_VERSION = 1.14.1
+AUTOMAKE_VERSION = 1.16.1
 AUTOMAKE_SOURCE = automake-$(AUTOMAKE_VERSION).tar.xz
 AUTOMAKE_SITE = $(BR2_GNU_MIRROR)/automake
 AUTOMAKE_LICENSE = GPLv2+

