diff --git a/package/Config.in b/package/Config.in
index 3bc8d24..a49709f 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -1,6 +1,7 @@
 menu "Target packages"
 
 source "package/busybox/Config.in"
+source "package/osmc/Config.in"
 
 menu "Audio and video applications"
 source "package/alsa-utils/Config.in"

diff --git a/package/osmc/Config.in b/package/osmc/Config.in
new file mode 100644
index 0000000..4c92bbc
--- /dev/null
+++ b/package/osmc/Config.in
@@ -0,0 +1,13 @@
+config BR2_PACKAGE_OSMC
+        bool "OSMC Installer"
+    select BR2_PACKAGE_QT
+    select BR2_PACKAGE_PARTED
+    select BR2_PACKAGE_XZ
+    select BR2_PACKAGE_DOSFSTOOLS
+    select BR2_PACKAGE_E2FSPROGS
+    select BR2_PACKAGE_PV
+    select BR2_PACKAGE_QT_LICENSE_APPROVED
+    select BR2_PACKAGE_FREETYPE
+    select BR2_PACKAGE_NETCAT
+        help
+          OSMC device side installer

diff --git a/package/osmc/osmc.mk b/package/osmc/osmc.mk
new file mode 100644
index 0000000..5158518
--- /dev/null
+++ b/package/osmc/osmc.mk
@@ -0,0 +1,25 @@
+#############################################################
+#
+# osmc installer
+#
+#############################################################
+
+
+OSMC_VERSION=1.0
+OSMC_SITE=$(TOPDIR)/../qt_target/qt_target_installer
+OSMC_SITE_METHOD=local
+OSMC_INSTALL_STAGING = NO
+OSMC_DEPENDENCIES = qt parted xz dosfstools e2fsprogs pv
+
+define OSMC_BUILD_CMDS
+	(cd $(@D) ; $(QT_QMAKE))
+	$(MAKE) -C $(@D) all
+	$(TARGET_STRIP) $(@D)/qt_target_installer
+endef
+
+define OSMC_INSTALL_TARGET_CMDS
+        $(INSTALL) -m 0755 $(@D)/qt_target_installer $(TARGET_DIR)/usr/bin/qt_target_installer
+        $(INSTALL) -m 0755 $(@D)/init $(TARGET_DIR)/init
+endef
+
+$(eval $(generic-package))
