--- a/package/qt/qt.mk	2021-10-10 14:12:50.952750453 +0100
+++ b/package/qt/qt.mk	2021-10-10 14:13:11.892999144 +0100
@@ -14,7 +14,7 @@
 QT_VERSION_MAJOR = 4.8
 QT_VERSION = $(QT_VERSION_MAJOR).6
 QT_SOURCE  = qt-everywhere-opensource-src-$(QT_VERSION).tar.gz
-QT_SITE    = http://download.qt-project.org/official_releases/qt/$(QT_VERSION_MAJOR)/$(QT_VERSION)
+QT_SITE    = http://download.osmc.tv/buildroot/2014.05
 QT_DEPENDENCIES = host-pkgconf
 QT_INSTALL_STAGING = YES
 
