--- a/package/Config.in	2023-04-16 17:16:56.972140606 +0100
+++ b/package/Config.in	2023-04-16 17:17:12.944328341 +0100
@@ -2,6 +2,7 @@
 
 source "package/busybox/Config.in"
 source "package/osmc/Config.in"
+source "package/osmcprovision/Config.in"
 
 menu "Audio and video applications"
 source "package/alsa-utils/Config.in"
 
--- /dev/null	2023-04-16 15:46:39.622993460 +0100
+++ b/package/osmcprovision/osmcprovision.mk	2023-04-16 17:15:31.807129709 +0100
@@ -0,0 +1,17 @@
+#############################################################
+#
+# osmc provison
+#
+#############################################################
+
+
+OSMCPROVISION_VERSION=1.0
+OSMCPROVISION_SITE=$(TOPDIR)/../provision
+OSMCPROVISION_SITE_METHOD=local
+OSMCPROVISION_INSTALL_STAGING = NO
+
+define OSMCPROVISION_INSTALL_TARGET_CMDS
+        $(INSTALL) -m 0755 $(@D)/init $(TARGET_DIR)/init
+endef
+
+$(eval $(generic-package))

--- /dev/null	2023-04-16 15:46:39.622993460 +0100
+++ b/package/osmcprovision/Config.in	2023-04-16 17:13:56.289973373 +0100
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_OSMCPROVISION
+        bool "OSMC PROVISION ENVIRONMENT"
+    select BR2_PACKAGE_PARTED
+    select BR2_PACKAGE_DOSFSTOOLS
+    select BR2_PACKAGE_E2FSPROGS
+    select BR2_PACKAGE_NETCAT
+        help
+          OSMC minimal provisoning environment

