diff --git a/package/Config.in b/package/Config.in
index 9e6de66..1885e27 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -1,6 +1,7 @@
 menu "Target packages"
 
 	source "package/osmc/Config.in"
+	source "package/osmcprovision/Config.in"
 	source "package/busybox/Config.in"
 	source "package/skeleton/Config.in"
 	source "package/skeleton-custom/Config.in"
diff --git a/package/osmcprovision/Config.in b/package/osmcprovision/Config.in
new file mode 100644
index 0000000..f03f349
--- /dev/null
+++ b/package/osmcprovision/Config.in
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_OSMCPROVISION
+        bool "OSMC PROVISION ENVIRONMENT"
+    select BR2_PACKAGE_PARTED
+    select BR2_PACKAGE_DOSFSTOOLS
+    select BR2_PACKAGE_E2FSPROGS
+    select BR2_PACKAGE_NETCAT
+        help
+          OSMC minimal provisoning environment
diff --git a/package/osmcprovision/osmcprovision.mk b/package/osmcprovision/osmcprovision.mk
new file mode 100644
index 0000000..3e33fe9
--- /dev/null
+++ b/package/osmcprovision/osmcprovision.mk
@@ -0,0 +1,17 @@
+#############################################################
+#
+# osmc provison
+#
+#############################################################
+
+
+OSMCPROVISION_VERSION=1.0
+OSMCPROVISION_SITE=$(TOPDIR)/../provision
+OSMCPROVISION_SITE_METHOD=local
+OSMCPROVISION_INSTALL_STAGING = NO
+
+define OSMCPROVISION_INSTALL_TARGET_CMDS
+        $(INSTALL) -m 0755 $(@D)/init $(TARGET_DIR)/init
+endef
+
+$(eval $(generic-package))
