--- a/package/rpi-firmware/rpi-firmware.mk      2014-05-31 08:52:49.000000000 +0100
+++ b/package/rpi-firmware/rpi-firmware.mk      2014-10-12 04:25:28.233268467 +0100
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = c8881991d9181779aee9637d6f4b215f108021a3
+RPI_FIRMWARE_VERSION = 216c2f79ca0633bebe3916184bf3550e1b8f2f80
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3c
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom

