diff --git a/package/rpi-firmware-osmc/Config.in b/package/rpi-firmware-osmc/Config.in
new file mode 100644
index 0000000..3bc006e
--- /dev/null
+++ b/package/rpi-firmware-osmc/Config.in
@@ -0,0 +1,48 @@
+config BR2_PACKAGE_RPI_FIRMWARE_OSMC
+	bool "package/rpi-firmware-osmc"
+	depends on BR2_arm
+	help
+	  RaspberryPi Firmware
+	  Pre-compiled binaries of the current bootloader and GPU firmware
+
+	  https://github.com/raspberrypi/firmware
+
+if BR2_PACKAGE_RPI_FIRMWARE_OSMC
+
+choice
+	bool "Firmware to boot"
+	default BR2_PACKAGE_RPI_FIRMWARE_OSMC_DEFAULT
+	help
+	  There are three different firmware files:
+	    - the default firmware, that enables standard GPU features;
+	    - the extended firmware, that enables additional GPU features
+	      (eg. more audio/video codecs);
+	    - the cut-down firmware, for emergency situations, with only
+	      features required to boot a Linux kernel.
+
+config BR2_PACKAGE_RPI_FIRMWARE_OSMC_DEFAULT
+	bool "default"
+	help
+	  The default firmware, that enables standard GPU features.
+
+config BR2_PACKAGE_RPI_FIRMWARE_OSMC_X
+	bool "extended ('x', more codecs)"
+	help
+	  The extended firmware, that enables additional GPU features
+	  (eg. more audio/video codecs).
+
+config BR2_PACKAGE_RPI_FIRMWARE_OSMC_CD
+	bool "cut-down ('cd', emergency)"
+	help
+	  The cut-down firmware, for emergency situations, with only
+	  features required to boot a Linux kernel.
+
+endchoice
+
+config BR2_PACKAGE_RPI_FIRMWARE_OSMC_BOOT
+	string
+	default ""      if BR2_PACKAGE_RPI_FIRMWARE_OSMC_DEFAULT
+	default "_x"    if BR2_PACKAGE_RPI_FIRMWARE_OSMC_X
+	default "_cd"   if BR2_PACKAGE_RPI_FIRMWARE_OSMC_CD
+
+endif # BR2_PACKAGE_RPI_FIRMWARE_OSMC
diff --git a/package/rpi-firmware-osmc/config.txt b/package/rpi-firmware-osmc/config.txt
new file mode 100644
index 0000000..8d6f2ed
--- /dev/null
+++ b/package/rpi-firmware-osmc/config.txt
@@ -0,0 +1,3 @@
+disable_overscan=1
+framebuffer_depth=32
+gpu_mem=16
diff --git a/package/rpi-firmware-osmc/rpi-firmware-osmc.mk b/package/rpi-firmware-osmc/rpi-firmware-osmc.mk
new file mode 100644
index 0000000..6a3a63b
--- /dev/null
+++ b/package/rpi-firmware-osmc/rpi-firmware-osmc.mk
@@ -0,0 +1,20 @@
+################################################################################
+#
+# package/rpi-firmware-osmc
+#
+################################################################################
+
+RPI_FIRMWARE_OSMC_VERSION = ed611df9c7f8dd1a274ba0a927100a4ee471f446
+RPI_FIRMWARE_OSMC_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_OSMC_VERSION))
+RPI_FIRMWARE_OSMC_LICENSE = BSD-3c
+RPI_FIRMWARE_OSMC_LICENSE_FILES = boot/LICENCE.broadcom
+
+define RPI_FIRMWARE_OSMC_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0644 $(@D)/boot/bootcode.bin $(BINARIES_DIR)/INSTALLER/bootcode.bin
+	$(INSTALL) -D -m 0644 $(@D)/boot/start_cd.elf $(BINARIES_DIR)/INSTALLER/start_cd.elf
+	$(INSTALL) -D -m 0644 $(@D)/boot/fixup_cd.dat $(BINARIES_DIR)/INSTALLER/fixup_cd.dat
+	$(INSTALL) -D -m 0644 package/rpi-firmware-osmc/config.txt $(BINARIES_DIR)/INSTALLER/config.txt
+	$(INSTALL) -D -m 0644 package/rpi-firmware-osmc/cmdline.txt $(BINARIES_DIR)/INSTALLER/cmdline.txt
+endef
+
+$(eval $(generic-package))
