--- a/linux/linux.mk	2018-03-13 00:07:54.961242027 +0000
+++ b/linux/linux.mk	2018-03-13 17:44:37.691351423 +0000
@@ -287,6 +287,10 @@
 define LINUX_INSTALL_IMAGES_CMDS
 	cp $(LINUX_IMAGE_PATH) $(BINARIES_DIR)
 	$(LINUX_INSTALL_DTB)
+        $(TARGET_MAKE_ENV) $(MAKE) $(LINUX_MAKE_FLAGS) -C $(@D) dtbs
+	mkdir -p $(BINARIES_DIR)/overlays
+	cp $(KERNEL_ARCH_PATH)/boot/dts/overlays/*.dtbo $(BINARIES_DIR)/overlays
+	cp $(KERNEL_ARCH_PATH)/boot/dts/*.dtb $(BINARIES_DIR)
 endef
 
 define LINUX_INSTALL_TARGET_CMDS
