--- a/linux/linux.mk	2021-03-17 21:29:27.386052266 +0000
+++ b/linux/linux.mk	2021-03-17 22:51:56.512510478 +0000
@@ -207,6 +207,7 @@
 	$(if $(BR2_LINUX_KERNEL_APPENDED_DTB),
 		$(call KCONFIG_ENABLE_OPT,CONFIG_ARM_APPENDED_DTB,$(@D)/.config))
 	yes '' | $(TARGET_MAKE_ENV) $(MAKE1) $(LINUX_MAKE_FLAGS) -C $(@D) oldconfig
+	cd $(@D); ARCH=arm64 scripts/kconfig/merge_config.sh .config arch/arm64/configs/enforce-kernel-module-signature.fragment
 endef
 
 ifeq ($(BR2_LINUX_KERNEL_DTS_SUPPORT),y)
--- a/Makefile	2023-04-04 16:33:08.664065620 +0100
+++ b/Makefile	2023-04-04 16:33:18.572279321 +0100
@@ -555,9 +555,6 @@
 endif
 	rm -rf $(TARGET_DIR)/usr/lib/luarocks
 	$(STRIP_FIND_CMD) | xargs $(STRIPCMD) 2>/dev/null || true
-	if test -d $(TARGET_DIR)/lib/modules; then \
-		find $(TARGET_DIR)/lib/modules -type f -name '*.ko' | \
-		xargs -r $(KSTRIPCMD); fi
 
 # See http://sourceware.org/gdb/wiki/FAQ, "GDB does not see any threads
 # besides the one in which crash occurred; or SIGTRAP kills my program when
--- a/linux/linux.mk	2023-04-16 17:33:36.570490587 +0100
+++ b/linux/linux.mk	2023-04-16 17:55:32.322247107 +0100
@@ -294,7 +294,7 @@
 	# Install modules and remove symbolic links pointing to build
 	# directories, not relevant on the target
 	@if grep -q "CONFIG_MODULES=y" $(@D)/.config; then 	\
-		$(TARGET_MAKE_ENV) $(MAKE1) $(LINUX_MAKE_FLAGS) -C $(@D) modules_install; \
+		$(TARGET_MAKE_ENV) $(MAKE1) $(LINUX_MAKE_FLAGS) -C $(@D) modules_install INSTALL_MOD_STRIP=1; \
 		rm -f $(TARGET_DIR)/lib/modules/$(LINUX_VERSION_PROBED)/build ;		\
 		rm -f $(TARGET_DIR)/lib/modules/$(LINUX_VERSION_PROBED)/source ;	\
 	fi
