diff --git a/linux/linux.mk b/linux/linux.mk
index 864fa77..c3f2102 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -382,10 +382,9 @@ endef
 ifeq ($(BR2_LINUX_KERNEL_APPENDED_DTB),)
 define LINUX_INSTALL_DTB
 	# dtbs moved from arch/<ARCH>/boot to arch/<ARCH>/boot/dts since 3.8-rc1
-	cp $(addprefix \
-		$(LINUX_ARCH_PATH)/boot/$(if $(wildcard \
-		$(addprefix $(LINUX_ARCH_PATH)/boot/dts/,$(LINUX_DTBS))),dts/),$(LINUX_DTBS)) \
-		$(1)
+        cp $(addprefix \
+                $(LINUX_ARCH_PATH)/boot/dts/amlogic/,$(LINUX_DTBS)) \
+                $(BINARIES_DIR)/
 endef
 endif # BR2_LINUX_KERNEL_APPENDED_DTB
 endif # BR2_LINUX_KERNEL_DTB_IS_SELF_BUILT
