--- a/linux/linux.mk	2016-01-23 15:20:46.107343396 +0000
+++ b/linux/linux.mk	2016-01-23 22:48:54.331375342 +0000
@@ -77,6 +77,7 @@
 endif
 endif
 
+KERNEL_DTDS = $(addsuffix .dtd,$(KERNEL_DTS_NAME))
 KERNEL_DTBS = $(addsuffix .dtb,$(KERNEL_DTS_NAME))
 
 ifeq ($(BR2_LINUX_KERNEL_IMAGE_TARGET_CUSTOM),y)
@@ -212,14 +213,14 @@
 ifeq ($(BR2_LINUX_KERNEL_DTS_SUPPORT),y)
 ifeq ($(BR2_LINUX_KERNEL_DTB_IS_SELF_BUILT),)
 define LINUX_BUILD_DTB
+	$(TARGET_MAKE_ENV) $(MAKE) $(LINUX_MAKE_FLAGS) -C $(@D) $(KERNEL_DTDS)
 	$(TARGET_MAKE_ENV) $(MAKE) $(LINUX_MAKE_FLAGS) -C $(@D) $(KERNEL_DTBS)
 endef
 define LINUX_INSTALL_DTB
 	# dtbs moved from arch/<ARCH>/boot to arch/<ARCH>/boot/dts since 3.8-rc1
-	cp $(addprefix \
-		$(KERNEL_ARCH_PATH)/boot/$(if $(wildcard \
-		$(addprefix $(KERNEL_ARCH_PATH)/boot/dts/,$(KERNEL_DTBS))),dts/),$(KERNEL_DTBS)) \
-		$(BINARIES_DIR)/
+        cp $(addprefix \
+                $(KERNEL_ARCH_PATH)/boot/dts/amlogic/,$(KERNEL_DTBS)) \
+                $(BINARIES_DIR)/
 endef
 define LINUX_INSTALL_DTB_TARGET
 	# dtbs moved from arch/<ARCH>/boot to arch/<ARCH>/boot/dts since 3.8-rc1
