diff -Naur linux/drivers/media/tuners/r820t.c linux/drivers/media/tuners/r820t.c
--- linux/drivers/media/tuners/r820t.c	2016-12-30 23:12:01.000000000 +0300
+++ linux/drivers/media/tuners/r820t.c	2016-12-30 23:15:48.000000000 +0300
@@ -766,7 +766,7 @@
 		lna_top = 0xe5;
 		lna_vth_l = 0x62;
 		mixer_vth_l = 0x75;
-		air_cable1_in = 0x60;
+		air_cable1_in = 0x00;
 		cable2_in = 0x00;
 		pre_dect = 0x40;
 		lna_discharge = 14;
