From e8b0754025ddd4a5a7c9b98f828c2d3c46e7f512 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 18 Apr 2015 01:48:52 +0100
Subject: [PATCH] smsc95xx: Disable turbo_mode by default

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 drivers/net/usb/smsc95xx.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/usb/smsc95xx.c b/drivers/net/usb/smsc95xx.c
index 5ae60ab..b0c3fda 100644
--- a/drivers/net/usb/smsc95xx.c
+++ b/drivers/net/usb/smsc95xx.c
@@ -71,9 +71,7 @@ struct smsc95xx_priv {
 	u8 suspend_flags;
 };
 
-static bool turbo_mode = true;
-module_param(turbo_mode, bool, 0644);
-MODULE_PARM_DESC(turbo_mode, "Enable multiple frames per Rx transaction");
+static bool turbo_mode = false;
 
 static char *macaddr = ":";
 module_param(macaddr, charp, 0);
-- 
2.1.0

