From a99187c46656a9abb341410b0bc598278818cd54 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 10 Mar 2018 16:04:20 +0000
Subject: [PATCH] Allow SMBv1 without vers=1

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 fs/cifs/connect.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/cifs/connect.c b/fs/cifs/connect.c
index 0bfc228..03651f4 100644
--- a/fs/cifs/connect.c
+++ b/fs/cifs/connect.c
@@ -1285,8 +1285,8 @@ cifs_parse_mount_options(const char *mountdata, const char *devname,
 	vol->actimeo = CIFS_DEF_ACTIMEO;
 
 	/* offer SMB2.1 and later (SMB3 etc). Secure and widely accepted */
-	vol->ops = &smb30_operations;
-	vol->vals = &smbdefault_values;
+        vol->ops = &smb1_operations;
+        vol->vals = &smb1_values;
 
 	vol->echo_interval = SMB_ECHO_INTERVAL_DEFAULT;
 
-- 
2.7.4

