From d46ab0c155285330ac8c4463b1470b2f364ce7c3 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 31 Jan 2018 17:34:46 +0000
Subject: [PATCH] Allow CFQ to be selected as default scheduler

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 block/Kconfig.iosched | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/block/Kconfig.iosched b/block/Kconfig.iosched
index a4a8914..d4e5e1a 100644
--- a/block/Kconfig.iosched
+++ b/block/Kconfig.iosched
@@ -57,6 +57,9 @@ choice
 	config DEFAULT_NOOP
 		bool "No-op"
 
+	config DEFAULT_BFQ
+		bool "BFQ" if IOSCHED_BFQ=y
+
 endchoice
 
 config DEFAULT_IOSCHED
@@ -64,6 +67,7 @@ config DEFAULT_IOSCHED
 	default "deadline" if DEFAULT_DEADLINE
 	default "cfq" if DEFAULT_CFQ
 	default "noop" if DEFAULT_NOOP
+	default "bfq" if DEFAULT_BFQ
 
 config MQ_IOSCHED_DEADLINE
 	tristate "MQ deadline I/O scheduler"
@@ -82,7 +86,7 @@ config MQ_IOSCHED_KYBER
 
 config IOSCHED_BFQ
 	tristate "BFQ I/O scheduler"
-	default n
+	default y
 	---help---
 	BFQ I/O scheduler for BLK-MQ. BFQ distributes the bandwidth of
 	of the device among all processes according to their weights,
-- 
2.7.4

