--- a/init/main.c	2020-11-16 23:49:16.355952562 +0000
+++ b/init/main.c	2020-11-16 23:49:52.844647981 +0000
@@ -1452,7 +1452,8 @@
 			return 0;
 	}
 
-	if (!try_to_run_init_process("/sbin/init") ||
+	if  (!try_to_run_init_process("/sbin/splash_early") ||
+            !try_to_run_init_process("/sbin/init") ||
 	    !try_to_run_init_process("/etc/init") ||
 	    !try_to_run_init_process("/bin/init") ||
 	    !try_to_run_init_process("/bin/sh"))
