--- a/sound/soc/codecs/pcm512x	2014-12-23 18:07:01.907480860 +0000
+++ b/sound/soc/codecs/pcm512x.c	2014-12-23 18:08:05.881861540 +0000
@@ -437,13 +437,11 @@
 	ret = regmap_write(regmap, PCM512x_RESET,
 			   PCM512x_RSTM | PCM512x_RSTR);
 	if (ret != 0) {
-		dev_err(dev, "Failed to reset device: %d\n", ret);
 		goto err;
 	}
 
 	ret = regmap_write(regmap, PCM512x_RESET, 0);
 	if (ret != 0) {
-		dev_err(dev, "Failed to reset device: %d\n", ret);
 		goto err;
 	}
 

