diff --git a/fs/nfs/nfsroot.c b/fs/nfs/nfsroot.c
index fa14830..6c10f5b 100644
--- a/fs/nfs/nfsroot.c
+++ b/fs/nfs/nfsroot.c
@@ -93,7 +93,7 @@
 #elif defined(CONFIG_NFS_V3)
 #define NFS_DEF_OPTIONS		"vers=3,tcp,rsize=4096,wsize=4096"
 #else
-#define NFS_DEF_OPTIONS		"vers=4,tcp,rsize=4096,wsize=4096"
+#define NFS_DEF_OPTIONS		"vers=3,tcp,rsize=4096,wsize=4096"
 #endif
 
 /* Parameters passed from the kernel command line */
