From 51ccd777fbd5733441c2daca081613d5f3f5badc Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Thu, 1 Feb 2018 01:06:45 +0000
Subject: [PATCH] LAN78xx: disable WOL as it can cause kernel panics

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 drivers/net/usb/lan78xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index da0b0bc..fc6df27 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -1619,7 +1619,7 @@ static const struct ethtool_ops lan78xx_ethtool_ops = {
 	.get_sset_count = lan78xx_get_sset_count,
 	.get_strings	= lan78xx_get_strings,
 	.get_wol	= lan78xx_get_wol,
-	.set_wol	= lan78xx_set_wol,
+//	.set_wol	= lan78xx_set_wol,
 	.get_eee	= lan78xx_get_eee,
 	.set_eee	= lan78xx_set_eee,
 	.get_pauseparam	= lan78xx_get_pause,
-- 
2.7.4

