diff --git a/arch/arm/mach-imx/busfreq-imx6.c b/arch/arm/mach-imx/busfreq-imx6.c
index 7a62444..1323356 100644
--- a/arch/arm/mach-imx/busfreq-imx6.c
+++ b/arch/arm/mach-imx/busfreq-imx6.c
@@ -925,11 +925,6 @@ static struct platform_driver busfreq_driver = {
 
 static int __init busfreq_init(void)
 {
-	if (platform_driver_register(&busfreq_driver) != 0)
-		return -ENODEV;
-
-	printk(KERN_INFO "Bus freq driver module loaded\n");
-
 	return 0;
 }
 
diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index 3c27907..a762b0f 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -713,6 +713,10 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	/* Set pll4_audio to a value that can derive 5K-88.2KHz and 8K-96KHz */
 	clk_set_rate(clk[pll4_audio_div], 541900800);
 
+	clk_set_rate(clk[pll2_pfd2_396m], 352000000);
+	clk_set_parent(clk[vpu_axi_sel], clk[pll2_pfd2_396m]);
+
+
 	/* Set initial power mode */
 	imx6q_set_lpm(WAIT_CLOCKED);
 
diff --git a/arch/arm/mach-imx/mach-imx6q.c b/arch/arm/mach-imx/mach-imx6q.c
index cf62ee0..52bde0b 100644
--- a/arch/arm/mach-imx/mach-imx6q.c
+++ b/arch/arm/mach-imx/mach-imx6q.c
@@ -364,6 +364,10 @@ static void __init imx6q_opp_init(void)
 		pr_warn("failed to init OPP table\n");
 		goto put_node;
 	}
+	if (dev_pm_opp_disable(cpu_dev, 396000000)) {
+		pr_warn("failed to disable 396MHz OPP\n");
+		pr_info("remove 396MHz OPP for VPU running at 352MHz!\n");
+	}
 
 	imx6q_opp_check_speed_grading(cpu_dev);
 
diff --git a/drivers/cpufreq/imx6-cpufreq.c b/drivers/cpufreq/imx6-cpufreq.c
index a2be777..9690b80 100644
--- a/drivers/cpufreq/imx6-cpufreq.c
+++ b/drivers/cpufreq/imx6-cpufreq.c
@@ -305,6 +305,11 @@ static int imx6_cpufreq_probe(struct platform_device *pdev)
 			unsigned long volt = be32_to_cpup(val++);
 			if (freq_table[j].frequency == freq) {
 				imx6_soc_volt[soc_opp_count++] = volt;
+			if (freq == 792000) {
+				pr_info("increase SOC/PU voltage for VPU352MHz\n");
+				imx6_soc_volt[soc_opp_count-1] = 1250000;
+				}
+	
 				break;
 			}
 		}
