--- a/plugins/Makefile.am	2020-09-25 16:48:15.989131477 +0100
+++ b/plugins/Makefile.am	2020-09-25 16:48:48.069625960 +0100
@@ -50,16 +50,6 @@
 
 endif
 
-if BUILD_FTDI
-plugin_LTLIBRARIES          += ftdi.la ftdix.la
-ftdi_la_SOURCES             = ftdi.c
-ftdi_la_LDFLAGS             = $(AM_LDFLAGS) @FTDI_LIBS@
-ftdi_la_CFLAGS              = $(AM_CFLAGS) @FTDI_CFLAGS@
-ftdix_la_SOURCES            = ftdix.c
-ftdix_la_LDFLAGS            = $(AM_LDFLAGS) @FTDI_LIBS@
-ftdix_la_CFLAGS             = $(AM_CFLAGS) @FTDI_CFLAGS@
-endif
-
 if BUILD_LIBPORTAUDIO
 plugin_LTLIBRARIES          += audio.la
 audio_la_SOURCES            = audio.c
@@ -67,28 +57,6 @@
                               $(portaudio_lib_other)  @forkpty@
 endif
 
-if BUILD_LIBALSA
-plugin_LTLIBRARIES          += audio_alsa.la
-audio_alsa_la_SOURCES       = audio_alsa.c
-audio_alsa_la_LDFLAGS       = $(AM_LDFLAGS) -lasound
-endif
-
-if BUILD_ALSA_SB_RC
-plugin_LTLIBRARIES          += alsa_usb.la
-alsa_usb_la_SOURCES         = alsa_usb.c
-alsa_usb_la_LDFLAGS         = $(AM_LDFLAGS) -lasound
-endif
-
-if BUILD_I2CUSER
-plugin_LTLIBRARIES          += i2cuser.la
-i2cuser_la_SOURCES          = i2cuser.c
-endif
-
-if BUILD_DSP
-plugin_LTLIBRARIES          += dsp.la
-dsp_la_SOURCES              = dsp.c
-endif
-
 plugin_LTLIBRARIES          += atwf83.la
 atwf83_la_SOURCES           = atwf83.c
 atwf83_la_LDFLAGS           = $(AM_LDFLAGS) -lpthread
