diff --git a/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp b/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
index c58c28a..bbea641 100644
--- a/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp
@@ -28,6 +28,7 @@
 #include "linux/RBP.h"
 #include "utils/StringUtils.h"
 #include "settings/Settings.h"
+#include <stdlib.h>
 
 #ifndef __VIDEOCORE4__
 #define __VIDEOCORE4__
@@ -101,6 +102,20 @@ void CEGLNativeTypeRaspberryPI::Initialize()
 
   m_DllBcmHost = new DllBcmHost;
   m_DllBcmHost->Load();
+  CLog::Log(LOGDEBUG, "Unbind vtcon0 to save resources");
+  FILE *fp;
+  fp = fopen("/sys/class/vtconsole/vtcon1/bind", "w");
+  if (fp != NULL)
+  {
+      fprintf(fp, "0");
+      fclose(fp);
+      CLog::Log(LOGDEBUG, "Reducing framebuffer resolution with fbset to save resources");
+      system("/bin/fbset -xres 1 -yres 1 -vxres 1 -vyres 1");
+  }
+ else
+ {
+      CLog::Log(LOGDEBUG, "Could not open vtcon1 for writing. You probably ran this independent of the watchdog!"); 
+ }
 #endif
 }

