diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 16bf81d..f2c5894 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2342,7 +2342,7 @@ void CApplication::Render()
   if (limitFrames || !flip)
   {
     if (!limitFrames)
-      singleFrameTime = 40; //if not flipping, loop at 25 fps
+      singleFrameTime = 15; //if not flipping, loop at 60+ fps
 
     unsigned int frameTime = now - m_lastFrameTime;
     if (frameTime < singleFrameTime)

