From 77b0aa9b4999c1187491f790baac7c7288d761cc Mon Sep 17 00:00:00 2001
From: smallint <tahoma@gmx.de>
Date: Mon, 19 Jan 2015 20:38:42 +0100
Subject: [PATCH 09/16] [imx] Fixed mode change after SysfsUtils port

---
 xbmc/windowing/egl/EGLNativeTypeIMX.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index 9893558..061f6e6 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -208,7 +208,7 @@ bool CEGLNativeTypeIMX::SetNativeResolution(const RESOLUTION_INFO &res)
   DestroyNativeWindow();
   DestroyNativeDisplay();
 
-  SysfsUtils::SetString("/sys/class/graphics/fb0/mode", res.strId);
+  SysfsUtils::SetString("/sys/class/graphics/fb0/mode", res.strId + "\n");
 
   CreateNativeDisplay();
 
-- 
2.1.0

