From 1d041653a315c2c573b3f603e22c54396e17cd72 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Wed, 11 Jan 2017 04:30:27 +0000
Subject: [PATCH 1/4] Allow multiple entries

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 tools/env/fw_env.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 39f7333..a76fb71 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -1453,6 +1453,7 @@ static int get_config (char *fname)
 	int rc;
 	char dump[128];
 	char *devname;
+	int fd_test;
 
 	fp = fopen (fname, "r");
 	if (fp == NULL)
@@ -1473,6 +1474,12 @@ static int get_config (char *fname)
 		if (rc < 3)
 			continue;
 
+		fd_test = open (devname, O_RDWR);
+		if (fd_test < 0)
+			continue;
+		if (close (fd_test))
+			continue;
+
 		DEVNAME(i) = devname;
 
 		if (rc < 4)
-- 
2.7.4

