---
 src/timeserver.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/timeserver.c b/src/timeserver.c
index f0d33e5..f39b144 100644
--- a/src/timeserver.c
+++ b/src/timeserver.c
@@ -187,8 +187,7 @@ GSList *__connman_timeserver_get_all(struct connman_service *service)
 	char **fallback_ts;
 	int index, i;
 
-	if (__connman_clock_timeupdates() == TIME_UPDATES_MANUAL)
-		return NULL;
+	return NULL;
 
 	service_ts_config = connman_service_get_timeservers_config(service);
 
-- 
2.1.4

