From b568aec0a1f676246bfad775b1356dffd623e2c0 Mon Sep 17 00:00:00 2001
From: Lars Op den Kamp <lars@opdenkamp.eu>
Date: Tue, 1 Dec 2015 10:46:50 +0100
Subject: [PATCH] switch from libplatform to p8-platform

---
 src/libcec/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libcec/CMakeLists.txt b/src/libcec/CMakeLists.txt
index 55aceca..9696264 100644
--- a/src/libcec/CMakeLists.txt
+++ b/src/libcec/CMakeLists.txt
@@ -19,6 +19,10 @@ if (SUPPORTS_CXX11)
 endif()
 
 find_package(p8-platform REQUIRED)
+if (p8-platform_VERSION VERSION_LESS 2.0)
+  message(FATAL_ERROR "p8-platform 2.0+ is required")
+endif()
+
 find_package(Threads REQUIRED)
 
 include_directories(${platform_INCLUDE_DIRS}
-- 
2.1.0

