From 8ef32691188e48f8cc7ad8af7b3131202f80ba92 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sun, 30 Apr 2017 09:35:11 +0100
Subject: [PATCH] Revert LG-Reconnect request

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 src/libcec/implementations/SLCommandHandler.cpp | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/libcec/implementations/SLCommandHandler.cpp b/src/libcec/implementations/SLCommandHandler.cpp
index 386ff39..636e8b4 100644
--- a/src/libcec/implementations/SLCommandHandler.cpp
+++ b/src/libcec/implementations/SLCommandHandler.cpp
@@ -55,8 +55,8 @@ using namespace P8PLATFORM;
 #define SL_COMMAND_POWER_ON             0x03
 #define SL_COMMAND_CONNECT_REQUEST      0x04
 #define SL_COMMAND_SET_DEVICE_MODE      0x05
-#define SL_COMMAND_REQUEST_RECONNECT    0x0b
 #define SL_COMMAND_REQUEST_POWER_STATUS 0xa0
+#define SL_COMMAND_UNKNOWN              0x0b
 
 #define LIB_CEC     m_busDevice->GetProcessor()->GetLib()
 #define ToString(p) LIB_CEC->ToString(p)
@@ -127,15 +127,13 @@ int CSLCommandHandler::HandleVendorCommand(const cec_command &command)
     return COMMAND_HANDLED;
   }
   else if (command.parameters.size == 1 &&
-      command.parameters[0] == SL_COMMAND_REQUEST_RECONNECT)
+      command.parameters[0] == SL_COMMAND_REQUEST_POWER_STATUS)
   {
-    HandleVendorCommandPowerOn(command);
+    HandleVendorCommandPowerOnStatus(command);
     return COMMAND_HANDLED;
   }
   else if (command.parameters.size == 1 &&
-      command.parameters[0] == SL_COMMAND_REQUEST_POWER_STATUS)
-  {
-    HandleVendorCommandPowerOnStatus(command);
+      command.parameters[0] == SL_COMMAND_UNKNOWN) {
     return COMMAND_HANDLED;
   }
 
-- 
2.7.4

