From a7ae8ff3a81724d480f2dcd76d2879c18cfb3466 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Thu, 1 Sep 2016 18:07:35 +0100
Subject: [PATCH] Remove ChangeLog target, we don't want to depend on Git

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 Makefile.am | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index edab5bb..4079296 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,15 +2,7 @@ ACLOCAL_AMFLAGS = -Im4
 
 AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_builddir)/src/dvdread $(CSS_CFLAGS)
 
-dist_doc_DATA = AUTHORS ChangeLog COPYING NEWS README TODO
-
-MAINTAINERCLEANFILES = ChangeLog
-
-ChangeLog: $(wildcard $(srcdir)/.git/logs/HEAD)
-	-cd $(srcdir) && git log > $(abs_builddir)/$(@)-tmp
-	-test -s $(@)-tmp && mv $(@)-tmp $(@)
-	-rm -f $(@)-tmp
-	test -e $(@) || touch $(@)
+dist_doc_DATA = AUTHORS COPYING NEWS README TODO
 
 EXTRA_DIST = msvc/ doc/footer.html doc/header.html
 
-- 
2.7.4

